=== modified file 'services/panel-main.c'
--- services/panel-main.c	2012-02-13 13:37:24 +0000
+++ services/panel-main.c	2012-08-02 15:46:27 +0000
@@ -235,7 +235,7 @@
 {
   GError *error = NULL;
   g_dbus_connection_emit_signal (connection,
-                                 S_NAME,
+                                 NULL,
                                  S_PATH,
                                  S_IFACE,
                                  "ReSync",
@@ -256,7 +256,7 @@
 {
   GError *error = NULL;
   g_dbus_connection_emit_signal (connection,
-                                 S_NAME,
+                                 NULL,
                                  S_PATH,
                                  S_IFACE,
                                  "EntryActivated",
@@ -278,7 +278,7 @@
   GError *error = NULL;
   g_warning ("%s, entry_id:%s", G_STRFUNC, entry_id);
   g_dbus_connection_emit_signal (connection,
-                                 S_NAME,
+                                 NULL,
                                  S_PATH,
                                  S_IFACE,
                                  "EntryActivateRequest",
@@ -300,7 +300,7 @@
 {
   GError *error = NULL;
   g_dbus_connection_emit_signal (connection,
-                                 S_NAME,
+                                 NULL,
                                  S_PATH,
                                  S_IFACE,
                                  "EntryShowNowChanged",

