=== modified file 'plugins/unityshell/src/unityshell.cpp'
--- plugins/unityshell/src/unityshell.cpp	2013-09-20 18:25:21 +0000
+++ plugins/unityshell/src/unityshell.cpp	2013-09-23 21:17:28 +0000
@@ -1344,18 +1344,19 @@
 
     if (action == ShowdesktopHandlerWindowInterface::PostPaintAction::Remove)
     {
+      wi->DeleteHandler();
       it = ShowdesktopHandler::animating_windows.erase(it);
       continue;
     }
     else if (action == ShowdesktopHandlerWindowInterface::PostPaintAction::Damage)
     {
-      wi->AddDamage ();
+      wi->AddDamage();
     }
 
     ++it;
   }
 
-  cScreen->donePaint ();
+  cScreen->donePaint();
 }
 
 void redraw_view_if_damaged(nux::ObjectPtr<nux::View> const& view, CompRegion const& damage)

