diff -urN unity-7.1.0+13.10.20130903.1/services/CMakeLists.txt unity-7.1.0+13.10.20130903.1-patched/services/CMakeLists.txt
--- unity-7.1.0+13.10.20130903.1/services/CMakeLists.txt	2013-09-07 17:13:08.793157223 +0930
+++ unity-7.1.0+13.10.20130903.1-patched/services/CMakeLists.txt	2013-09-07 17:21:22.877883055 +0930
@@ -12,10 +12,6 @@
     indicator3-0.4>=12.10.2
     x11
     libido3-0.1>=13.0.0
-    libupstart
-    libnih
-    libnih-dbus
-    dbus-1
 )
 
 pkg_check_modules(SERVICE_DEPS REQUIRED ${UNITY_PANEL_SERVICE_DEPS})
diff -urN unity-7.1.0+13.10.20130903.1/services/panel-service.c unity-7.1.0+13.10.20130903.1-patched/services/panel-service.c
--- unity-7.1.0+13.10.20130903.1/services/panel-service.c	2013-09-03 16:06:49.000000000 +0930
+++ unity-7.1.0+13.10.20130903.1-patched/services/panel-service.c	2013-09-07 17:25:45.086588948 +0930
@@ -32,9 +32,6 @@
 #include <X11/extensions/XInput2.h>
 #include <X11/XKBlib.h>
 
-#include <upstart.h>
-#include <nih/alloc.h>
-
 G_DEFINE_TYPE (PanelService, panel_service, G_TYPE_OBJECT);
 
 #define GET_PRIVATE(o) \
@@ -79,7 +76,6 @@
   IndicatorObjectEntry *pressed_entry;
   gboolean use_event;
 
-  NihDBusProxy * upstart;
 };
 
 /* Globals */
@@ -154,17 +150,6 @@
   g_idle_remove_by_data (self);
   gdk_window_remove_filter (NULL, (GdkFilterFunc)event_filter, self);
 
-  if (priv->upstart != NULL) 
-    {
-      int event_sent = 0;
-      event_sent = upstart_emit_event_sync (NULL, priv->upstart, "indicator-services-end", NULL, 0);
-      if (event_sent != 0)
-         g_warning("Unable to signal for indicator services to start");
-
-      nih_unref (priv->upstart, NULL);
-      priv->upstart = NULL;
-    }
-
   if (GTK_IS_WIDGET (priv->menubar) &&
       gtk_widget_get_realized (GTK_WIDGET (priv->menubar)))
     {
@@ -483,12 +468,35 @@
 static gboolean
 ready_signal (PanelService *self)
 {
-  if (PANEL_IS_SERVICE (self) && self->priv->upstart != NULL)
+  if (PANEL_IS_SERVICE (self))
     {
-      int event_sent = 0;
-      event_sent = upstart_emit_event_sync (NULL, self->priv->upstart, "indicator-services-start", NULL, 0);
-      if (event_sent != 0)
-         g_warning ("Unable to signal for indicator services to start");
+      GError * error = NULL;
+       gchar * argv[] = {
+         "initctl",
+         "--session",
+         "--user",
+         "emit",
+         "--no-wait",
+         "indicators-loaded",
+         NULL,
+       };
+  
+       g_spawn_async (NULL, /* Working Directory */
+                      argv,
+                      NULL, /* environment */
+                      G_SPAWN_SEARCH_PATH,
+                      NULL, NULL, /* child setup function */
+                      NULL, /* Pid */
+                      &error);
+  
+       if (error)
+               {
+           /* NOTE: When we get to the point where we can start
+              assuming upstart user sessions this can be escillated
+              to a warning or higher */
+           g_debug ("Unable to signal indicators-loaded to upstart: %s", error->message);
+           g_error_free (error);
+         }
     }
 
   return FALSE;
@@ -583,25 +591,6 @@
                     G_CALLBACK (on_keybinding_changed), self);
 
   panel_service_update_menu_keybinding (self);
-
-  const gchar * upstartsession = g_getenv ("UPSTART_SESSION");
-  if (upstartsession != NULL)
-    {
-      DBusConnection * conn = NULL;
-      conn = dbus_connection_open (upstartsession, NULL);
-      if (conn != NULL)
-        {
-          priv->upstart = nih_dbus_proxy_new (NULL, conn,
-                                              NULL,
-                                              DBUS_PATH_UPSTART,
-                                              NULL, NULL);
-          dbus_connection_unref (conn);
-        }
-    }
-
-  if (priv->upstart != NULL)
-    priv->upstart->auto_start = FALSE;
-
 }
 
 static gboolean
diff -urN unity-7.1.0+13.10.20130903.1/tests/CMakeLists.txt unity-7.1.0+13.10.20130903.1-patched/tests/CMakeLists.txt
--- unity-7.1.0+13.10.20130903.1/tests/CMakeLists.txt	2013-09-07 17:13:08.798157291 +0930
+++ unity-7.1.0+13.10.20130903.1-patched/tests/CMakeLists.txt	2013-09-07 17:21:57.582354867 +0930
@@ -36,7 +36,7 @@
 #
 # Unit tests
 #
-set (TEST_DEPS "${UNITY_PLUGIN_DEPS};unity>=4.99.0;libupstart;libnih;libnih-dbus;dbus-1")
+set (TEST_DEPS "${UNITY_PLUGIN_DEPS};unity>=4.99.0")
 pkg_check_modules (TEST_UNIT_DEPS REQUIRED ${TEST_DEPS})
 
 string (REPLACE ";" " " TEST_UNIT_DEPS_CFLAGS_OTHER "${TEST_UNIT_CFLAGS_OTHER}")
