From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Thu, 13 Dec 2018 18:00:42 -0500
Subject: print-dialog-show-options-of-remote-dnssd-printers

Bug-Upstream: https://bugzilla.gnome.org/show_bug.cgi?id=684533
---
 modules/printbackends/cups/gtkprintbackendcups.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

Index: gtk3/modules/printbackends/cups/gtkprintbackendcups.c
===================================================================
--- gtk3.orig/modules/printbackends/cups/gtkprintbackendcups.c
+++ gtk3/modules/printbackends/cups/gtkprintbackendcups.c
@@ -2513,6 +2513,19 @@ cups_create_printer (GtkPrintBackendCups
 		   &port,
 		   resource, sizeof (resource));
 
+  if (strncmp (method, "dnssd", 5) == 0)
+    {
+      _httpResolveURI(cups_printer->printer_uri,
+		      uri, sizeof(uri), 0, NULL, NULL);
+      httpSeparateURI (HTTP_URI_CODING_ALL,
+		       uri,
+		       method, sizeof (method),
+		       username, sizeof (username),
+		       hostname, sizeof (hostname),
+		       &port,
+		       resource, sizeof (resource));
+    }
+
   if (strncmp (resource, "/printers/", 10) == 0)
     {
       cups_printer->ppd_name = g_strdup (resource + 10);
