diff -urN lttng-tools-2.6.0/src/bin/lttng/commands/enable_events.c lttng-tools-2.6.0-patched/src/bin/lttng/commands/enable_events.c
--- lttng-tools-2.6.0/src/bin/lttng/commands/enable_events.c	2015-01-27 03:47:15.000000000 +1030
+++ lttng-tools-2.6.0-patched/src/bin/lttng/commands/enable_events.c	2016-03-06 10:20:12.057508331 +1030
@@ -50,7 +50,6 @@
 static int opt_enable_all;
 static char *opt_probe;
 static char *opt_function;
-static char *opt_function_entry_symbol;
 static char *opt_channel_name;
 static char *opt_filter;
 static char *opt_exclude;
@@ -983,11 +982,13 @@
 					goto error;
 				}
 				break;
+#if 0
 			case LTTNG_EVENT_FUNCTION_ENTRY:
 				strncpy(ev.attr.ftrace.symbol_name, opt_function_entry_symbol,
 						LTTNG_SYMBOL_NAME_LEN);
 				ev.attr.ftrace.symbol_name[LTTNG_SYMBOL_NAME_LEN - 1] = '\0';
 				break;
+#endif
 			case LTTNG_EVENT_SYSCALL:
 				ev.type = LTTNG_EVENT_SYSCALL;
 				break;
