diff -urN indicator-appmenu-0.3.97/src/hud-cli.c indicator-appmenu-0.3.97-patched/src/hud-cli.c
--- indicator-appmenu-0.3.97/src/hud-cli.c	2012-04-06 01:33:45.000000000 +0930
+++ indicator-appmenu-0.3.97-patched//src/hud-cli.c	2012-05-25 10:55:30.215311281 +0930
@@ -19,6 +19,7 @@
 with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <string.h>
 #include <glib.h>
 #include <gio/gunixinputstream.h>
 #include <gtk/gtk.h>
