From: Benjamin Berg <bberg@redhat.com>
Date: Mon, 19 Oct 2020 12:06:55 +0200
Subject: main: Use already defined error from main function.

Origin: upstream, 40, commit:33bc7b8bb5a4d2dbeaa8d3298b66bc8c75fc8adb
---
 gnome-session/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnome-session/main.c b/gnome-session/main.c
index 84edfbe..53f174e 100644
--- a/gnome-session/main.c
+++ b/gnome-session/main.c
@@ -539,7 +539,6 @@ main (int argc, char **argv)
 
 #ifdef ENABLE_SYSTEMD_SESSION
         if (use_systemd && !systemd_service) {
-                g_autoptr(GError) error = NULL;
                 g_autofree gchar *gnome_session_target;
                 const gchar *session_type;
 
@@ -571,6 +570,7 @@ main (int argc, char **argv)
 
                 /* We could not start the unit, fall back. */
                  g_warning ("Falling back to non-systemd startup procedure due to error: %s", error->message);
+                 g_clear_error (&error);
         }
 #endif /* ENABLE_SYSTEMD_SESSION */
 
