diff -Naur gnome-screensaver-3.6.1_orig/src/gs-manager.c gnome-screensaver-3.6.1/src/gs-manager.c
--- gnome-screensaver-3.6.1_orig/src/gs-manager.c	2012-08-20 19:12:39.000000000 +0200
+++ gnome-screensaver-3.6.1/src/gs-manager.c	2014-01-12 16:32:44.359489000 +0100
@@ -582,9 +582,13 @@
 
         g_settings_delay (settings);
 
+        /* do not reset 'picture-uri' and 'picture-options' to show user specific background.
+         */
         keys = g_settings_list_keys (settings);
         for (k = keys; *k; k++) {
-                g_settings_reset (settings, *k);
+                if(g_strcmp0(*k, "picture-uri") && g_strcmp0(*k, "picture-uri")) {
+                        g_settings_reset (settings, *k);
+                }
         }
         g_strfreev (keys);
 
