Index: unity-greeter-12.10.4/src/dash-entry.vala
===================================================================
--- unity-greeter-12.10.4.orig/src/dash-entry.vala	2012-10-08 11:30:03.000000000 -0400
+++ unity-greeter-12.10.4/src/dash-entry.vala	2012-10-08 11:32:37.757222564 -0400
@@ -175,6 +175,20 @@
         show_arrow (get_text_length () != 0);
     }
 
+    public override bool key_press_event (Gdk.EventKey event)
+    {
+        // This is a workaroud for bug https://launchpad.net/bugs/944159
+        // The problem is that orca seems to not notice that it's in a password
+        // field on startup.  We just need to kick orca in the pants.
+        if (UnityGreeter.singleton.orca_needs_kick)
+        {
+            Signal.emit_by_name (get_accessible (), "focus-event", true);
+            UnityGreeter.singleton.orca_needs_kick = false;
+        }
+
+        return base.key_press_event (event);
+    }
+
     protected void show_arrow (bool visible)
     {
         if (visible && can_respond && !did_respond)
Index: unity-greeter-12.10.4/src/menubar.vala
===================================================================
--- unity-greeter-12.10.4.orig/src/menubar.vala	2012-10-08 11:30:03.000000000 -0400
+++ unity-greeter-12.10.4/src/menubar.vala	2012-10-08 11:32:37.761222565 -0400
@@ -487,6 +487,23 @@
                                      SpawnFlags.SEARCH_PATH,
                                      null,
                                      out reader_pid);
+                // This is a workaroud for bug https://launchpad.net/bugs/944159
+                // The problem is that orca seems to not notice that it's in a
+                // password field on startup.  We just need to kick orca in the
+                // pants.  We do this two ways:  a racy way and a non-racy way.
+                // We kick it after a second which is ideal if we win the race,
+                // because the user gets to hear what widget they are in, and
+                // the first character will be masked.  Otherwise, if we lose
+                // that race, the first time the user types (see
+                // DashEntry.key_press_event), we will kick orca again.  While
+                // this is not racy with orca startup, it is racy with whether
+                // orca wll read the first character or not out loud.  Hence
+                // why we do both.  Ideally this would be fixed in orca itself.
+                UnityGreeter.singleton.orca_needs_kick = true;
+                Timeout.add_seconds (1, () => {
+                    Signal.emit_by_name ((get_toplevel () as Gtk.Window).get_focus ().get_accessible (), "focus-event", true);
+                    return false;
+                });
             }
             catch (Error e)
             {
Index: unity-greeter-12.10.4/src/unity-greeter.vala
===================================================================
--- unity-greeter-12.10.4.orig/src/unity-greeter.vala	2012-10-08 11:30:03.653220934 -0400
+++ unity-greeter-12.10.4/src/unity-greeter.vala	2012-10-08 11:32:37.761222565 -0400
@@ -37,6 +37,7 @@
 
     private SettingsDaemon settings_daemon;
 
+    public bool orca_needs_kick;
     private MainWindow main_window;
 
     private LightDM.Greeter greeter;
Index: unity-greeter-12.10.4/tests/unity-greeter.vala
===================================================================
--- unity-greeter-12.10.4.orig/tests/unity-greeter.vala	2012-10-08 11:30:03.653220934 -0400
+++ unity-greeter-12.10.4/tests/unity-greeter.vala	2012-10-08 11:32:37.765222566 -0400
@@ -30,6 +30,7 @@
     public bool test_mode = false;
     public bool session_started = false;
     public string last_respond_response;
+    public bool orca_needs_kick;
 
     public bool is_authenticated ()
     {
