Index: unity-greeter-12.10.4/src/unity-greeter.vala
===================================================================
--- unity-greeter-12.10.4.orig/src/unity-greeter.vala	2012-10-09 10:29:59.429703123 -0400
+++ unity-greeter-12.10.4/src/unity-greeter.vala	2012-10-09 10:30:59.313703757 -0400
@@ -58,6 +58,7 @@
         greeter = new LightDM.Greeter ();
         greeter.show_message.connect ((text, type) => { show_message (text, type); });
         greeter.show_prompt.connect ((text, type) => { show_prompt (text, type); });
+        greeter.autologin_timer_expired.connect (() => { greeter.authenticate_autologin (); });
         greeter.authentication_complete.connect (() => { authentication_complete (); });
         var connected = false;
         try
