=== modified file 'src/unity-greeter.vala'
Index: unity-greeter-12.10.3/src/unity-greeter.vala
===================================================================
--- unity-greeter-12.10.3.orig/src/unity-greeter.vala	2012-09-20 15:35:40.803122424 -0400
+++ unity-greeter-12.10.3/src/unity-greeter.vala	2012-09-20 15:36:12.755122762 -0400
@@ -452,21 +452,6 @@
         if (do_test_mode)
             debug ("Running in test mode");
 
-        if (!do_test_mode)
-        {
-            /* Make nm-applet hide items the user does not have permissions to interact with */
-            Environment.set_variable ("NM_APPLET_HIDE_POLICY_ITEMS", "1", true);
-
-            try
-            {
-                Process.spawn_command_line_async ("nm-applet");
-            }
-            catch (Error e)
-            {
-                warning ("Error starting nm-applet: %s", e.message);
-            }
-        }
-
         /* Set GTK+ settings */
         debug ("Setting GTK+ settings");
         var settings = Gtk.Settings.get_default ();
@@ -497,6 +482,21 @@
         debug ("Showing greeter");
         greeter.show ();
 
+        if (!do_test_mode)
+        {
+            /* Make nm-applet hide items the user does not have permissions to interact with */
+            Environment.set_variable ("NM_APPLET_HIDE_POLICY_ITEMS", "1", true);
+
+            try
+            {
+                Process.spawn_command_line_async ("nm-applet");
+            }
+            catch (Error e)
+            {
+                warning ("Error starting nm-applet: %s", e.message);
+            }
+        }
+
         debug ("Starting main loop");
         Gtk.main ();
 
