Index: unity-greeter-12.10.4/src/menubar.vala
===================================================================
--- unity-greeter-12.10.4.orig/src/menubar.vala	2012-10-01 10:01:32.000000000 -0400
+++ unity-greeter-12.10.4/src/menubar.vala	2012-10-03 10:16:56.428792116 -0400
@@ -174,7 +174,7 @@
     {
         if (keyboard_pid != 0)
         {
-            Posix.kill (keyboard_pid, Posix.SIGKILL);
+            Posix.kill (keyboard_pid, Posix.SIGTERM);
             int status;
             Posix.waitpid (keyboard_pid, out status, 0);
             keyboard_pid = 0;
