Description: Include Software Center in menus

Index: gnome-menus-3.36.0/layout/gnome-applications.menu
===================================================================
--- gnome-menus-3.36.0.orig/layout/gnome-applications.menu
+++ gnome-menus-3.36.0/layout/gnome-applications.menu
@@ -450,4 +450,16 @@
      <Merge type="files" />
    </Layout>
 
+<Include>
+  <Filename>org.gnome.Software.desktop</Filename>
+</Include>
+
+<!-- Separator between menus and gnome-app-install -->
+<Layout>
+  <Merge type="menus"/>
+  <Merge type="files"/>
+  <Separator/>
+  <Filename>org.gnome.Software.desktop</Filename>
+</Layout>
+
 </Menu> <!-- End Applications -->
