Index: gnome-menus-3.35.3/layout/gnome-applications.menu
===================================================================
--- gnome-menus-3.35.3.orig/layout/gnome-applications.menu
+++ gnome-menus-3.35.3/layout/gnome-applications.menu
@@ -97,8 +97,107 @@
     <Include>
       <And>
         <Category>Game</Category>
-      </And>
-    </Include>
+        <Not><Category>ActionGame</Category></Not>
+        <Not><Category>AdventureGame</Category></Not>
+        <Not><Category>ArcadeGame</Category></Not>
+        <Not><Category>BoardGame</Category></Not>
+        <Not><Category>BlocksGame</Category></Not>
+        <Not><Category>CardGame</Category></Not>
+        <Not><Category>KidsGame</Category></Not>
+        <Not><Category>LogicGame</Category></Not>
+        <Not><Category>Simulation</Category></Not>
+        <Not><Category>SportsGame</Category></Not>
+        <Not><Category>StrategyGame</Category></Not>
+      </And>
+    </Include>
+    <DefaultLayout inline="true" inline_limit="6" inline_header="false">
+      <Merge type="menus"/>
+      <Merge type="files"/>
+    </DefaultLayout>
+    <Menu>
+      <Name>Action</Name>
+      <Directory>ActionGames.directory</Directory>
+      <Include>
+        <Category>ActionGame</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Adventure</Name>
+      <Directory>AdventureGames.directory</Directory>
+      <Include>
+        <Category>AdventureGame</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Arcade</Name>
+      <Directory>ArcadeGames.directory</Directory>
+      <Include>
+        <Category>ArcadeGame</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Board</Name>
+      <Directory>BoardGames.directory</Directory>
+      <Include>
+        <Category>BoardGame</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Blocks</Name>
+      <Directory>BlocksGames.directory</Directory>
+      <Include>
+        <Category>BlocksGame</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Cards</Name>
+      <Directory>CardGames.directory</Directory>
+      <Include>
+        <Category>CardGame</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Kids</Name>
+      <Directory>KidsGames.directory</Directory>
+      <Include>
+        <Category>KidsGame</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Logic</Name>
+      <Directory>LogicGames.directory</Directory>
+      <Include>
+        <Category>LogicGame</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Role Playing</Name>
+      <Directory>RolePlayingGames.directory</Directory>
+      <Include>
+        <Category>RolePlaying</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Simulation</Name>
+      <Directory>SimulationGames.directory</Directory>
+      <Include>
+        <Category>Simulation</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Sports</Name>
+      <Directory>SportsGames.directory</Directory>
+      <Include>
+        <Category>SportsGame</Category>
+      </Include>
+    </Menu>
+    <Menu>
+      <Name>Strategy</Name>
+      <Directory>StrategyGames.directory</Directory>
+      <Include>
+        <Category>StrategyGame</Category>
+      </Include>
+    </Menu>
   </Menu> <!-- End Games -->
 
   <!-- Graphics -->
@@ -175,6 +274,7 @@
     <Include>
       <And>
         <Category>System</Category>
+        <Not><Category>Game</Category></Not>
         <Not><Category>Settings</Category></Not>
         <Not><Category>X-GNOME-Utilities</Category></Not>
       </And>
