diff -urN metacity-2.34.1/src/metacity.schemas.in.in metacity-2.34.1-patched/src/metacity.schemas.in.in
--- metacity-2.34.1/src/metacity.schemas.in.in	2012-08-22 15:25:21.944450335 +0930
+++ metacity-2.34.1-patched/src/metacity.schemas.in.in	2012-08-22 15:27:02.329455195 +0930
@@ -55,7 +55,7 @@
       <applyto>/apps/metacity/general/show_maximized_titlebars</applyto>
       <owner>metacity</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
          <short>Whether to show title bars for maximized windows</short>
          <long>
