diff -urN metacity-3.36.1/src/compositor/meta-compositor-vulkan.c metacity-3.36.1-patched/src/compositor/meta-compositor-vulkan.c
--- metacity-3.36.1/src/compositor/meta-compositor-vulkan.c	2020-03-07 03:34:14.000000000 +1030
+++ metacity-3.36.1-patched/src/compositor/meta-compositor-vulkan.c	2020-06-23 20:43:47.259372476 +0930
@@ -810,7 +810,9 @@
         return "cpu";
         break;
 
+#if VK_HEADER_VERSION < 140
       case VK_PHYSICAL_DEVICE_TYPE_RANGE_SIZE:
+#endif
       case VK_PHYSICAL_DEVICE_TYPE_MAX_ENUM:
       default:
         break;
