Description: Port to enchant-2
Author: Laurent Bigonville <bigon@debian.org>
Forwarded: yes

--- a/configure.ac
+++ b/configure.ac
@@ -306,7 +306,7 @@ AC_ARG_ENABLE(spell,
 if test "x$enable_spell" != "xno"; then
    PKG_CHECK_MODULES(ENCHANT,
    [
-      enchant >= $ENCHANT_REQUIRED,
+      enchant-2 >= $ENCHANT_REQUIRED,
       iso-codes >= $ISO_CODES_REQUIRED
    ], have_enchant="yes", have_enchant="no")
 
--- a/libempathy-gtk/empathy-spell.c
+++ b/libempathy-gtk/empathy-spell.c
@@ -424,7 +424,7 @@ empathy_spell_add_to_dictionary (const g
 	if (lang == NULL)
 		return;
 
-	enchant_dict_add_to_pwl (lang->speller, word, strlen (word));
+	enchant_dict_add (lang->speller, word, strlen (word));
 }
 
 #else /* not HAVE_ENCHANT */
