diff -Naur qt-everywhere-opensource-src-4.8.1.orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h qt-everywhere-opensource-src-4.8.1/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h
--- qt-everywhere-opensource-src-4.8.1.orig/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h	2012-04-01 19:32:46.621395314 +0200
+++ qt-everywhere-opensource-src-4.8.1/src/3rdparty/webkit/Source/JavaScriptCore/wtf/gobject/GOwnPtr.h	2012-04-01 19:34:23.693181749 +0200
@@ -24,6 +24,7 @@

 #if ENABLE(GLIB_SUPPORT)

+#include <glib.h>
 #include <algorithm>
 #include <wtf/Assertions.h>
 #include <wtf/Noncopyable.h>
