diff -urN googletest-release-1.8.1-patched/googlemock/include/gmock/internal/gmock-internal-utils.h googletest-release-1.8.1/googlemock/include/gmock/internal/gmock-internal-utils.h
--- googletest-release-1.8.1-patched/googlemock/include/gmock/internal/gmock-internal-utils.h	2020-01-08 23:19:44.739750362 +1030
+++ googletest-release-1.8.1/googlemock/include/gmock/internal/gmock-internal-utils.h	2020-01-08 23:21:15.457013988 +1030
@@ -125,8 +125,11 @@
 //
 // To gcc,
 //   wchar_t == signed wchar_t != unsigned wchar_t == unsigned int
+//
+// gcc-9 appears to treat signed/unsigned wchar_t as ill-formed
+// regardless of the signage of its underlying type.
 #ifdef __GNUC__
-#if !defined(__WCHAR_UNSIGNED__)
+#if !defined(__WCHAR_UNSIGNED__) && (__GNUC__ < 9)
 // signed/unsigned wchar_t are valid types.
 # define GMOCK_HAS_SIGNED_WCHAR_T_ 1
 #endif
