Description: Use Ubuntu help instead of GNOME help

Index: gnome-screenshot-3.3.91/src/screenshot-utils.c
===================================================================
--- gnome-screenshot-3.3.91.orig/src/screenshot-utils.c	2012-03-05 16:48:12.000000000 +0100
+++ gnome-screenshot-3.3.91/src/screenshot-utils.c	2012-03-13 20:48:21.073565050 +0100
@@ -713,7 +713,7 @@
   GError *error = NULL;
 
   gtk_show_uri (gtk_window_get_screen (parent),
-		"help:gnome-help/screen-shot-record",
+		"help:ubuntu-help/screen-shot-record",
 		gtk_get_current_event_time (), &error);
 
   if (error)
