diff -Nur a/protocol/protocol-icon.vala b/protocol/protocol-icon.vala
--- a/protocol/protocol-icon.vala	2019-03-19 11:17:56.000000000 +0100
+++ b/protocol/protocol-icon.vala	2021-06-08 10:03:32.978507708 +0200
@@ -185,7 +185,7 @@
   }
 
   /* Added to GIcon interface in 2.37 */
-  private Variant serialize ()
+  private Variant? serialize ()
   {
     Variant? ret = null;
     return ret;
diff -Nur a/protocol/protocol-preview-player.vala b/protocol/protocol-preview-player.vala
--- a/protocol/protocol-preview-player.vala	2019-03-19 11:17:56.000000000 +0100
+++ b/protocol/protocol-preview-player.vala	2021-06-08 10:03:25.765495002 +0200
@@ -38,8 +38,8 @@
  */
 public class PreviewPlayer: GLib.Object
 {
-  static const string PREVIEW_PLAYER_DBUS_NAME = "com.canonical.Unity.Lens.Music.PreviewPlayer";
-  static const string PREVIEW_PLAYER_DBUS_PATH = "/com/canonical/Unity/Lens/Music/PreviewPlayer";
+  const string PREVIEW_PLAYER_DBUS_NAME = "com.canonical.Unity.Lens.Music.PreviewPlayer";
+  const string PREVIEW_PLAYER_DBUS_PATH = "/com/canonical/Unity/Lens/Music/PreviewPlayer";
 
   /**
    * Reports progress of playback for given track uri.
diff -Nur a/protocol/protocol-scope-discovery.vala b/protocol/protocol-scope-discovery.vala
--- a/protocol/protocol-scope-discovery.vala	2019-03-19 11:17:56.000000000 +0100
+++ b/protocol/protocol-scope-discovery.vala	2021-06-08 10:03:25.765495002 +0200
@@ -21,7 +21,7 @@
 namespace Unity.Protocol
 {
 
-  private static const string SCOPES_DIR = "unity/scopes";
+  private const string SCOPES_DIR = "unity/scopes";
 
   errordomain ParseError
   {
@@ -187,8 +187,8 @@
 
   public class ScopeRegistry
   {
-    private static const string SCOPE_GROUP = "Scope";
-    private static const string DESKTOP_GROUP = "Desktop Entry";
+    private const string SCOPE_GROUP = "Scope";
+    private const string DESKTOP_GROUP = "Desktop Entry";
 
     private static string[] scope_directories = null;
     private static string[] scope_file_prefixes = null;
@@ -899,7 +899,7 @@
 
   public class ScopeGroupConfig
   {
-    private static const string SCOPE_GROUP_GROUP = "Scope Group";
+    private const string SCOPE_GROUP_GROUP = "Scope Group";
 
     public List<ScopeGroupScopeInfo?> scopes;
     public int timeout;
