diff -Nur unity-chromium-extension-3.2.0+15.04.20150119.orig/messaging-host/main.cpp unity-chromium-extension-3.2.0+15.04.20150119/messaging-host/main.cpp
--- unity-chromium-extension-3.2.0+15.04.20150119.orig/messaging-host/main.cpp	2015-01-19 17:49:27.000000000 +0100
+++ unity-chromium-extension-3.2.0+15.04.20150119/messaging-host/main.cpp	2016-06-07 09:37:30.933692735 +0200
@@ -45,7 +45,10 @@
     case QtFatalMsg:
         syslog_type |= LOG_ERR;
         break;
-    }
+    case QtInfoMsg:
+        syslog_type |= LOG_INFO;
+        break;
+    }    
 
     syslog(syslog_type, "%s", localMsg.constData());
 }
