diff -urN qtbase-opensource-src-5.2.0-beta1/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in qtbase-opensource-src-5.2.0-beta1-patched/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in
--- qtbase-opensource-src-5.2.0-beta1/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2013-10-22 19:19:28.000000000 +1030
+++ qtbase-opensource-src-5.2.0-beta1-patched/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2013-10-29 07:19:45.863404117 +1030
@@ -185,7 +185,9 @@
         list(APPEND Qt5$${CMAKE_MODULE_NAME}_EXECUTABLE_COMPILE_FLAGS ${Qt5${_module_dep}_EXECUTABLE_COMPILE_FLAGS})
     endforeach()
     list(REMOVE_DUPLICATES Qt5$${CMAKE_MODULE_NAME}_INCLUDE_DIRS)
-    list(REMOVE_DUPLICATES Qt5$${CMAKE_MODULE_NAME}_PRIVATE_INCLUDE_DIRS)
+    if (Qt5$${CMAKE_MODULE_NAME}_PRIVATE_INCLUDE_DIRS)
+        list(REMOVE_DUPLICATES Qt5$${CMAKE_MODULE_NAME}_PRIVATE_INCLUDE_DIRS)
+    endif()
     list(REMOVE_DUPLICATES Qt5$${CMAKE_MODULE_NAME}_DEFINITIONS)
     list(REMOVE_DUPLICATES Qt5$${CMAKE_MODULE_NAME}_COMPILE_DEFINITIONS)
     if (Qt5$${CMAKE_MODULE_NAME}_EXECUTABLE_COMPILE_FLAGS)
