--- bamf-0.4.0daily13.02.06/lib/libbamf/libbamf3.pc.in	2013-02-16 08:22:48.688755037 +1030
+++ bamf-0.3.4/lib/libbamf/libbamf3.pc.in	2013-02-16 08:22:39.926754612 +1030
@@ -7,7 +7,7 @@
 Description: C Convenience library for bamf d-bus server.
 Version: @VERSION@
 
-Requires: glib-2.0
+Requires: glib-2.0 libwnck-3.0
 Libs: -L${libdir} -lbamf3
-Cflags: -I${includedir}/libbamf3 -I${libdir}/libbamf3/include
+Cflags: -I${includedir}/libbamf3 -I${libdir}/libbamf3/include -DWNCK_I_KNOW_THIS_IS_UNSTABLE
 
