diff -Ncr clutter-gst-1.4.6-o/clutter-gst/Makefile.am clutter-gst-1.4.6/clutter-gst/Makefile.am
*** clutter-gst-1.4.6-o/clutter-gst/Makefile.am	2012-01-18 21:06:12.000000000 +0400
--- clutter-gst-1.4.6/clutter-gst/Makefile.am	2012-11-09 23:53:36.051190365 +0400
***************
*** 65,71 ****
  	-DCLUTTER_GST_COMPILATION		\
  	-DG_DISABLE_SINGLE_INCLUDES		\
  	-DCOGL_ENABLE_EXPERIMENTAL_API		\
- 	-DCOGL_ENABLE_EXPERIMENTAL_2_0_API	\
  	-DCLUTTER_ENABLE_EXPERIMENTAL_API	\
  	-DG_LOG_DOMAIN=\"Clutter-Gst\"		\
  	$(NULL)
--- 65,70 ----
