diff -urN trust-store-1.0.0+14.10.20140826.1/src/core/trust/mir/prompt_main.cpp trust-store-1.0.0+14.10.20140826.1-patched/src/core/trust/mir/prompt_main.cpp
--- trust-store-1.0.0+14.10.20140826.1/src/core/trust/mir/prompt_main.cpp	2014-08-27 00:40:00.000000000 +0930
+++ trust-store-1.0.0+14.10.20140826.1-patched/src/core/trust/mir/prompt_main.cpp	2014-09-22 07:44:25.495445814 +0930
@@ -19,6 +19,8 @@
  *  Thomas Voß <thomas.voss@canonical.com>
  */
 
+#include <iostream>
+
 // Qt
 #include <QCommandLineParser>
 #include <QGuiApplication>
