diff -urN hud-13.04.0daily13.04.03/src/hud-cli.c hud-13.04.0daily13.04.03-patched/src/hud-cli.c
--- hud-13.04.0daily13.04.03/src/hud-cli.c	2013-04-03 12:32:02.000000000 +1030
+++ hud-13.04.0daily13.04.03-patched/src/hud-cli.c	2013-04-13 08:33:27.493103709 +0930
@@ -19,6 +19,7 @@
 with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <string.h>
 #include <glib.h>
 #include <gio/gunixinputstream.h>
 #include <gtk/gtk.h>
