Author: Jamie Strandboge <jamie@canonical.com>
Description: increase CONTAINER_MAX_TRACKS to 15000
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=750642
Bug-Ubuntu: https://launchpad.net/bugs/1548883

Index: rhythmbox-3.4.3/plugins/grilo/rb-grilo-source.c
===================================================================
--- rhythmbox-3.4.3.orig/plugins/grilo/rb-grilo-source.c
+++ rhythmbox-3.4.3/plugins/grilo/rb-grilo-source.c
@@ -51,7 +51,7 @@
 /* maximum number of tracks to fetch before stopping and
  * requiring the user to ask for more.
  */
-#define CONTAINER_MAX_TRACKS		1000
+#define CONTAINER_MAX_TRACKS		15000
 
 /* number of items to fetch at once */
 #define CONTAINER_FETCH_SIZE		50
