Index: rhythmbox-3.4.3/data/rhythmbox.desktop.in.in
===================================================================
--- rhythmbox-3.4.3.orig/data/rhythmbox.desktop.in.in
+++ rhythmbox-3.4.3/data/rhythmbox.desktop.in.in
@@ -18,7 +18,7 @@ X-GNOME-Bugzilla-Component=general
 X-GNOME-Bugzilla-OtherBinaries=rhythmbox-client;rhythmbox-metadata;
 X-GNOME-Bugzilla-Version=@VERSION@
 X-GNOME-UsesNotifications=true
-Actions=PlayPause;Next;Previous;
+Actions=PlayPause;Next;Previous;StopQuit;
 
 [Desktop Action PlayPause]
 _Name=Play/Pause
@@ -31,3 +31,7 @@ Exec=rhythmbox-client --next
 [Desktop Action Previous]
 _Name=Previous
 Exec=rhythmbox-client --previous
+
+[Desktop Action StopQuit]
+_Name=Stop & Quit
+Exec=rhythmbox-client --quit
