From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Tue, 13 Sep 2016 23:17:34 -0400
Subject: [PATCH] Load plugins from non-multiarch directory also

---
 shell/rb-shell.c | 3 +++
 1 file changed, 3 insertions(+)

Index: rhythmbox/shell/rb-shell.c
===================================================================
--- rhythmbox.orig/shell/rb-shell.c
+++ rhythmbox/shell/rb-shell.c
@@ -885,6 +885,9 @@ construct_plugins (RBShell *shell)
 	peas_engine_add_search_path (shell->priv->plugin_engine,
 				     plugindir,
 				     plugindatadir);
+	peas_engine_add_search_path (shell->priv->plugin_engine,
+				     "/usr/lib/rhythmbox/plugins",
+				     plugindatadir);
 	g_free (plugindir);
 	g_free (plugindatadir);
 
