diff -Naur signon-8.52daily13.06.16_orig/common-project-config.pri signon-8.52daily13.06.16/common-project-config.pri
--- signon-8.52daily13.06.16_orig/common-project-config.pri	2013-06-16 06:31:52.000000000 +0200
+++ signon-8.52daily13.06.16/common-project-config.pri	2013-06-30 10:46:36.566284387 +0200
@@ -3,7 +3,7 @@
 #-----------------------------------------------------------------------------
 
 QT		-= gui
-CONFIG         += link_pkgconfig
+CONFIG         += link_pkgconfig nostrip
 #MOC_DIR         = .moc
 #OBJECTS_DIR     = .obj
 RCC_DIR         = resources
