diff -Nur a/configure.ac b/configure.ac
--- a/configure.ac	2017-03-01 14:41:34.000000000 +0100
+++ b/configure.ac	2017-07-24 05:02:52.107025962 +0200
@@ -46,7 +46,7 @@
 PKG_CHECK_MODULES([FCITX_GCLIENT], [fcitx-gclient])
 PKG_CHECK_MODULES([ACCOUNTSSERVICE], [accountsservice])
 PKG_CHECK_MODULES([LIGHTDM], [liblightdm-gobject-1])
-PKG_CHECK_MODULES([URLDISPATCHER], [url-dispatcher-1 >= 1])
+#PKG_CHECK_MODULES([URLDISPATCHER], [url-dispatcher-1 >= 1])
 
 SYSTEMD_USERDIR=`$PKG_CONFIG --variable=systemduserunitdir systemd`
 AC_SUBST(SYSTEMD_USERDIR)
diff -Nur a/deps/url-dispatcher.vapi b/deps/url-dispatcher.vapi
--- a/deps/url-dispatcher.vapi	2017-03-01 14:41:34.000000000 +0100
+++ b/deps/url-dispatcher.vapi	2017-07-24 17:43:20.598784925 +0200
@@ -1,4 +1,4 @@
-[CCode (cprefix="", lower_case_cprefix="", cheader_filename="liburl-dispatcher-1/url-dispatcher.h")]
+/*[CCode (cprefix="", lower_case_cprefix="", cheader_filename="liburl-dispatcher-1/url-dispatcher.h")]
 
 namespace UrlDispatch
 {
@@ -6,4 +6,4 @@
 
   [CCode (cname = "url_dispatch_send")]
   public static void send (string url, DispatchCallback? func = null);
-}
+}*/
diff -Nur a/lib/main.vala b/lib/main.vala
--- a/lib/main.vala	2017-03-01 14:41:55.000000000 +0100
+++ b/lib/main.vala	2017-07-24 04:59:27.688069398 +0200
@@ -1201,7 +1201,7 @@
 	[DBus (visible = false)]
 	private void handle_activate_settings (Variant? parameter) {
 		if (Environment.get_variable ("MIR_SOCKET") != null) {
-			UrlDispatch.send ("settings:///system/hw-keyboard-layouts");
+//			UrlDispatch.send ("settings:///system/hw-keyboard-layouts");
 			return;
 		}
 
diff -Nur a/lib/Makefile.am b/lib/Makefile.am
--- a/lib/Makefile.am	2017-03-01 14:41:34.000000000 +0100
+++ b/lib/Makefile.am	2017-07-24 17:59:54.208983400 +0200
@@ -1,7 +1,7 @@
 pkglibexec_PROGRAMS = indicator-keyboard-service
 
 AM_CFLAGS = -w -DGNOME_DESKTOP_USE_UNSTABLE_API
-AM_LDFLAGS = -lm -lurl-dispatcher
+AM_LDFLAGS = -lm
 AM_VALAFLAGS = --enable-experimental-non-null   \
                --metadatadir $(top_srcdir)/deps \
                --vapidir $(top_srcdir)/deps
@@ -28,8 +28,8 @@
                                        --pkg ibus-1.0             \
                                        --pkg Fcitx-1.0            \
                                        --pkg AccountsService-1.0  \
-                                       --pkg liblightdm-gobject-1 \
-                                       --pkg url-dispatcher
+                                       --pkg liblightdm-gobject-1
+#                                       --pkg url-dispatcher
 indicator_keyboard_service_CFLAGS = $(AM_CFLAGS)              \
                                     $(GEE_CFLAGS)             \
                                     $(PANGOCAIRO_CFLAGS)      \
@@ -41,8 +41,8 @@
                                     $(FCITX_GCLIENT_CFLAGS)   \
                                     $(ACCOUNTSSERVICE_CFLAGS) \
                                     $(LIGHTDM_CFLAGS)         \
-                                    $(COVERAGE_CFLAGS)        \
-                                    $(URLDISPATCHER_CFLAGS)
+                                    $(COVERAGE_CFLAGS)
+#                                    $(URLDISPATCHER_CFLAGS)
 indicator_keyboard_service_LDFLAGS = $(AM_LDFLAGS)           \
                                      $(GEE_LIBS)             \
                                      $(PANGOCAIRO_LIBS)      \
@@ -54,5 +54,5 @@
                                      $(FCITX_GCLIENT_LIBS)   \
                                      $(ACCOUNTSSERVICE_LIBS) \
                                      $(LIGHTDM_LIBS)         \
-                                     $(COVERAGE_LDFLAGS)     \
-                                     $(URLDISPATCHER_LDFLAGS)
+                                     $(COVERAGE_LDFLAGS)
+#                                     $(URLDISPATCHER_LDFLAGS)
