diff -urN indicator-datetime-13.10.0+14.10.20140919/data/com.canonical.indicator.datetime.gschema.xml.in indicator-datetime-13.10.0+14.10.20140919-patched/data/com.canonical.indicator.datetime.gschema.xml.in
--- indicator-datetime-13.10.0+14.10.20140919/data/com.canonical.indicator.datetime.gschema.xml.in	2014-09-20 00:03:46.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/data/com.canonical.indicator.datetime.gschema.xml.in	2014-09-20 19:38:09.766241145 +0930
@@ -98,7 +98,7 @@
 			<default>false</default>
 			<_summary>Show the auto-detected location in the indicator</_summary>
 			<_description>
-			  Shows your current location (determined from geoclue and /etc/timezone) in indicator-datetime's menu.
+			  Shows your current location (determined from geoclue and systemd#s timedated) in indicator-datetime's menu.
 			</_description>
 		</key>
 		<key name="show-locations" type="b">
diff -urN indicator-datetime-13.10.0+14.10.20140919/include/datetime/timezone-file.h indicator-datetime-13.10.0+14.10.20140919-patched/include/datetime/timezone-file.h
--- indicator-datetime-13.10.0+14.10.20140919/include/datetime/timezone-file.h	2014-09-20 00:04:32.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/include/datetime/timezone-file.h	2014-09-20 19:38:09.767241145 +0930
@@ -36,6 +36,7 @@
 public:
     FileTimezone(const std::string& filename);
     ~FileTimezone();
+    static bool have_file(const std::string& filename);
 
 private:
     class Impl;
diff -urN indicator-datetime-13.10.0+14.10.20140919/include/datetime/timezones-live.h indicator-datetime-13.10.0+14.10.20140919-patched/include/datetime/timezones-live.h
--- indicator-datetime-13.10.0+14.10.20140919/include/datetime/timezones-live.h	2014-09-20 00:03:46.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/include/datetime/timezones-live.h	2014-09-20 19:38:09.767241145 +0930
@@ -24,6 +24,7 @@
 #include <datetime/timezones.h>
 #include <datetime/timezone-file.h>
 #include <datetime/timezone-geoclue.h>
+#include <datetime/timezone-systemd.h>
 
 #include <memory> // shared_ptr<>
 
@@ -43,8 +44,10 @@
 private:
     void update_geolocation();
     void update_timezones();
+    bool m_etc_timezone = false;
 
     FileTimezone m_file;
+    SystemdTimezone m_systemd;
     std::shared_ptr<const Settings> m_settings;
     std::shared_ptr<GeoclueTimezone> m_geo;
 };
diff -urN indicator-datetime-13.10.0+14.10.20140919/include/datetime/timezone-systemd.h indicator-datetime-13.10.0+14.10.20140919-patched/include/datetime/timezone-systemd.h
--- indicator-datetime-13.10.0+14.10.20140919/include/datetime/timezone-systemd.h	1970-01-01 09:30:00.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/include/datetime/timezone-systemd.h	2014-09-20 19:38:09.767241145 +0930
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2014 Xiao-Long Chen <chenxiaolong@cxl.epac.to>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3, as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranties of
+ * MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *   Charles Kerr <charles.kerr@canonical.com>
+ */
+
+#ifndef INDICATOR_DATETIME_SYSTEMD_TIMEZONE_H
+#define INDICATOR_DATETIME_SYSTEMD_TIMEZONE_H
+
+#include <datetime/timezone.h> // base class
+
+#include <string> // std::string
+
+#include "timedate1.h"
+#include <glib.h>
+#include <gio/gio.h>
+
+namespace unity {
+namespace indicator {
+namespace datetime {
+
+/**
+ * \brief A #Timezone that gets its information from systemd
+ */
+class SystemdTimezone : public Timezone
+{
+public:
+    SystemdTimezone();
+    ~SystemdTimezone();
+
+private:
+    static void on_timezone_changed(gpointer gself);
+    void set_timezone();
+
+    GCancellable * m_cancellable = nullptr;
+    Timedate1 * m_timedate = nullptr;
+
+    // we have raw pointers and glib tags in here, so disable copying
+    SystemdTimezone(const SystemdTimezone&) =delete;
+    SystemdTimezone& operator=(const SystemdTimezone&) =delete;
+};
+
+} // namespace datetime
+} // namespace indicator
+} // namespace unity
+
+#endif // INDICATOR_DATETIME_SYSTEMD_TIMEZONE_H
diff -urN indicator-datetime-13.10.0+14.10.20140919/src/CMakeLists.txt indicator-datetime-13.10.0+14.10.20140919-patched/src/CMakeLists.txt
--- indicator-datetime-13.10.0+14.10.20140919/src/CMakeLists.txt	2014-09-20 00:03:46.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/src/CMakeLists.txt	2014-09-20 19:38:09.767241145 +0930
@@ -4,6 +4,21 @@
 add_definitions (-DTIMEZONE_FILE="/etc/timezone"
                  -DG_LOG_DOMAIN="Indicator-Datetime")
 
+# This interface xml can only be generated when the timedated daemon and DBus
+# are available
+#gdbus introspect                                       \
+#       --xml                                           \
+#       --system                                        \
+#       --dest org.freedesktop.timedate1                \
+#       --object-path /org/freedesktop/timedate1        \
+#       > timedate1-interface.xml
+
+add_custom_command (
+    OUTPUT timedate1.h timedate1.c
+    COMMAND gdbus-codegen --interface-prefix org.freedesktop. --generate-c-code timedate1 ${CMAKE_CURRENT_SOURCE_DIR}/timedate1-interface.xml
+    COMMENT "Generating code for systemd's timedate1 interface"
+)
+
 # handwritten sources
 set (SERVICE_C_SOURCES
      utils.c)
@@ -36,8 +51,11 @@
      sound.cpp
      timezone-file.cpp
      timezone-geoclue.cpp
+     timezone-systemd.cpp
      timezones-live.cpp
      utils.c
+     timedate1.h
+     timedate1.c
      wakeup-timer-mainloop.cpp
      wakeup-timer-powerd.cpp)
 
diff -urN indicator-datetime-13.10.0+14.10.20140919/src/main.cpp indicator-datetime-13.10.0+14.10.20140919-patched/src/main.cpp
--- indicator-datetime-13.10.0+14.10.20140919/src/main.cpp	2014-09-20 00:04:11.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/src/main.cpp	2014-09-21 08:05:08.302387058 +0930
@@ -33,6 +33,7 @@
 #include <datetime/state.h>
 #include <datetime/timezone-file.h>
 #include <datetime/timezones-live.h>
+#include <datetime/timezone-systemd.h>
 #include <datetime/wakeup-timer-powerd.h>
 #include <notifications/notifications.h>
 
@@ -127,7 +128,8 @@
     textdomain(GETTEXT_PACKAGE);
 
     auto engine = create_engine();
-    auto timezone_ = std::make_shared<FileTimezone>(TIMEZONE_FILE);
+    std::shared_ptr<Timezone> timezone_ = (nullptr);
+    timezone_ = std::shared_ptr<Timezone>(new SystemdTimezone());
     auto state = create_state(engine, timezone_);
     auto actions = std::make_shared<LiveActions>(state);
     MenuFactory factory(actions, state);
diff -urN indicator-datetime-13.10.0+14.10.20140919/src/timedate1-interface.xml indicator-datetime-13.10.0+14.10.20140919-patched/src/timedate1-interface.xml
--- indicator-datetime-13.10.0+14.10.20140919/src/timedate1-interface.xml	1970-01-01 09:30:00.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/src/timedate1-interface.xml	2014-09-20 19:38:09.767241145 +0930
@@ -0,0 +1,74 @@
+<!DOCTYPE node PUBLIC "-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd">
+<node>
+ <interface name="org.freedesktop.DBus.Peer">
+  <method name="Ping"/>
+  <method name="GetMachineId">
+   <arg type="s" name="machine_uuid" direction="out"/>
+  </method>
+ </interface>
+ <interface name="org.freedesktop.DBus.Introspectable">
+  <method name="Introspect">
+   <arg name="data" type="s" direction="out"/>
+  </method>
+ </interface>
+ <interface name="org.freedesktop.DBus.Properties">
+  <method name="Get">
+   <arg name="interface" direction="in" type="s"/>
+   <arg name="property" direction="in" type="s"/>
+   <arg name="value" direction="out" type="v"/>
+  </method>
+  <method name="GetAll">
+   <arg name="interface" direction="in" type="s"/>
+   <arg name="properties" direction="out" type="a{sv}"/>
+  </method>
+  <method name="Set">
+   <arg name="interface" direction="in" type="s"/>
+   <arg name="property" direction="in" type="s"/>
+   <arg name="value" direction="in" type="v"/>
+  </method>
+  <signal name="PropertiesChanged">
+   <arg type="s" name="interface"/>
+   <arg type="a{sv}" name="changed_properties"/>
+   <arg type="as" name="invalidated_properties"/>
+  </signal>
+ </interface>
+ <interface name="org.freedesktop.timedate1">
+  <property name="Timezone" type="s" access="read">
+  </property>
+  <property name="LocalRTC" type="b" access="read">
+  </property>
+  <property name="CanNTP" type="b" access="read">
+   <annotation name="org.freedesktop.DBus.Property.EmitsChangedSignal" value="false"/>
+  </property>
+  <property name="NTP" type="b" access="read">
+  </property>
+  <property name="NTPSynchronized" type="b" access="read">
+   <annotation name="org.freedesktop.DBus.Property.EmitsChangedSignal" value="false"/>
+  </property>
+  <property name="TimeUSec" type="t" access="read">
+   <annotation name="org.freedesktop.DBus.Property.EmitsChangedSignal" value="false"/>
+  </property>
+  <property name="RTCTimeUSec" type="t" access="read">
+   <annotation name="org.freedesktop.DBus.Property.EmitsChangedSignal" value="false"/>
+  </property>
+  <method name="SetTime">
+   <arg type="x" direction="in"/>
+   <arg type="b" direction="in"/>
+   <arg type="b" direction="in"/>
+  </method>
+  <method name="SetTimezone">
+   <arg type="s" direction="in"/>
+   <arg type="b" direction="in"/>
+  </method>
+  <method name="SetLocalRTC">
+   <arg type="b" direction="in"/>
+   <arg type="b" direction="in"/>
+   <arg type="b" direction="in"/>
+  </method>
+  <method name="SetNTP">
+   <arg type="b" direction="in"/>
+   <arg type="b" direction="in"/>
+  </method>
+ </interface>
+</node>
diff -urN indicator-datetime-13.10.0+14.10.20140919/src/timezone-file.cpp indicator-datetime-13.10.0+14.10.20140919-patched/src/timezone-file.cpp
--- indicator-datetime-13.10.0+14.10.20140919/src/timezone-file.cpp	2014-09-20 00:04:32.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/src/timezone-file.cpp	2014-09-20 19:38:09.768241145 +0930
@@ -39,7 +39,8 @@
     Impl(FileTimezone& owner, const std::string& filename):
         m_owner(owner)
     {
-        set_filename(filename);
+        if (have_file(filename))
+            set_filename(filename);
     }
 
     ~Impl()
@@ -183,6 +184,12 @@
 {
 }
 
+bool
+FileTimezone::have_file(const std::string& filename)
+{
+    return g_file_test(filename.c_str(), G_FILE_TEST_EXISTS);
+}
+
 /***
 ****
 ***/
diff -urN indicator-datetime-13.10.0+14.10.20140919/src/timezones-live.cpp indicator-datetime-13.10.0+14.10.20140919-patched/src/timezones-live.cpp
--- indicator-datetime-13.10.0+14.10.20140919/src/timezones-live.cpp	2014-09-20 00:03:46.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/src/timezones-live.cpp	2014-09-20 19:38:09.768241145 +0930
@@ -28,9 +28,11 @@
 LiveTimezones::LiveTimezones(const std::shared_ptr<const Settings>& settings,
                              const std::string& filename):
     m_file(filename),
+    m_systemd(),
     m_settings(settings)
 {
-    m_file.timezone.changed().connect([this](const std::string&){update_timezones();});
+    g_warning("Getting timezone from systemd's timedate1 interface");
+    m_systemd.timezone.changed().connect([this](const std::string&){update_timezones();});
 
     m_settings->show_detected_location.changed().connect([this](bool){update_geolocation();});
     update_geolocation();
@@ -54,7 +56,8 @@
 
 void LiveTimezones::update_timezones()
 {
-    const auto a = m_file.timezone.get();
+    const auto tz = m_systemd.timezone;
+    const auto a = tz.get();
     const auto b = m_geo ? m_geo->timezone.get() : "";
 
     timezone.set(a.empty() ? b : a);
diff -urN indicator-datetime-13.10.0+14.10.20140919/src/timezone-systemd.cpp indicator-datetime-13.10.0+14.10.20140919-patched/src/timezone-systemd.cpp
--- indicator-datetime-13.10.0+14.10.20140919/src/timezone-systemd.cpp	1970-01-01 09:30:00.000000000 +0930
+++ indicator-datetime-13.10.0+14.10.20140919-patched/src/timezone-systemd.cpp	2014-09-20 19:38:09.768241145 +0930
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2014 Xiao-Long Chen <chenxiaolong@cxl.epac.to>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3, as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranties of
+ * MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <datetime/timezone-systemd.h>
+
+namespace unity {
+namespace indicator {
+namespace datetime {
+
+SystemdTimezone::SystemdTimezone() : m_cancellable(g_cancellable_new())
+{
+    GError * error = nullptr;
+    m_timedate = timedate1_proxy_new_for_bus_sync(G_BUS_TYPE_SYSTEM,
+                                                  G_DBUS_PROXY_FLAGS_NONE,
+                                                  "org.freedesktop.timedate1",
+                                                  "/org/freedesktop/timedate1",
+                                                  m_cancellable,
+                                                  &error);
+
+    if (!m_timedate)
+    {
+        g_warning("Could not get proxy for timedate1: %s", error->message);
+        g_clear_error(&error);
+        return;
+    }
+
+    set_timezone();
+
+    g_signal_connect_swapped(m_timedate,
+                             "notify::timezone",
+                             G_CALLBACK(on_timezone_changed),
+                             this);
+}
+
+SystemdTimezone::~SystemdTimezone()
+{
+    g_cancellable_cancel(m_cancellable);
+    g_clear_object(&m_cancellable);
+    g_clear_object(&m_timedate);
+}
+
+void
+SystemdTimezone::on_timezone_changed(gpointer gself)
+{
+    static_cast<SystemdTimezone*>(gself)->set_timezone();
+}
+
+void
+SystemdTimezone::set_timezone()
+{
+    const gchar * tz_new = timedate1_get_timezone(m_timedate);
+    timezone.set(tz_new);
+}
+
+} // namespace datetime
+} // namespace indicator
+} // namespace unity
