Index: cairo/src/Makefile.am
===================================================================
--- cairo.orig/src/Makefile.am	2011-02-08 10:11:46.879538772 +0100
+++ cairo/src/Makefile.am	2011-02-08 10:12:36.381915666 +0100
@@ -46,7 +46,7 @@
 	$(enabled_cairo_private) \
 	$(enabled_cairo_sources) \
 	$(NULL)
-libcairo_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols)
+libcairo_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols) -export-symbols-regex \^cairo_.*
 libcairo_la_LIBADD = $(CAIRO_LIBS) \
 	$(cairo_cxx_lib)
 libcairo_la_DEPENDENCIES = $(cairo_def_dependency) $(cairo_cxx_lib)
diff --git a/util/cairo-gobject/Makefile.am b/util/cairo-gobject/Makefile.am
index 22c1a27..5a15950 100644
--- a/util/cairo-gobject/Makefile.am
+++ b/util/cairo-gobject/Makefile.am
@@ -10,6 +10,6 @@ libcairo_gobject_la_SOURCES = \
 	$(NULL)
 
 libcairo_gobject_la_CFLAGS = $(CAIRO_CFLAGS) $(GOBJECT_CFLAGS)
-libcairo_gobject_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols)
+libcairo_gobject_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols) -export-symbols-regex \^cairo_.*
 libcairo_gobject_la_LIBADD = $(top_builddir)/src/libcairo.la $(CAIRO_LIBS) $(GOBJECT_LIBS)
 
diff --git a/util/cairo-script/Makefile.am b/util/cairo-script/Makefile.am
index d7e4427..8d4181b 100644
--- a/util/cairo-script/Makefile.am
+++ b/util/cairo-script/Makefile.am
@@ -18,7 +18,7 @@ libcairo_script_interpreter_la_SOURCES = \
 	cairo-script-stack.c \
 	$(NULL)
 libcairo_script_interpreter_la_CFLAGS = $(CAIRO_CFLAGS)
-libcairo_script_interpreter_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols)
+libcairo_script_interpreter_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols) -export-symbols-regex \^cairo_.*
 libcairo_script_interpreter_la_LIBADD = $(top_builddir)/src/libcairo.la $(CAIRO_LIBS) -lz
 
 csi_replay_SOURCES = csi-replay.c
