diff -urN url-dispatcher/CMakeLists.txt url-dispatcher-patched/CMakeLists.txt
--- url-dispatcher/CMakeLists.txt	2017-03-29 04:58:11.000000000 +1030
+++ url-dispatcher-patched/CMakeLists.txt	2017-06-05 09:19:14.621335751 +0930
@@ -43,10 +43,10 @@
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wpedantic -Wextra")
 endif()
 
-if (${werror})
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
-  set(CMAKE_C_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
-endif()
+#if (${werror})
+#  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+#  set(CMAKE_C_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+#endif()
 
 pkg_check_modules(UBUNTU_APP_LAUNCH REQUIRED ubuntu-app-launch-3>=0.5)
 include_directories(${UBUNTU_APP_LAUNCH_INCLUDE_DIRS})
