diff -urN chromium-45.0.2454.101/third_party/boringssl/boringssl.gyp chromium-45.0.2454.101-patched/third_party/boringssl/boringssl.gyp
--- chromium-45.0.2454.101/third_party/boringssl/boringssl.gyp	2015-09-25 18:36:24.000000000 +0930
+++ chromium-45.0.2454.101-patched/third_party/boringssl/boringssl.gyp	2015-10-21 21:38:06.270536703 +1030
@@ -17,6 +17,7 @@
       'defines': [
         'BORINGSSL_IMPLEMENTATION',
         'BORINGSSL_NO_STATIC_INITIALIZER',
+        '_POSIX_C_SOURCE=200909L',
       ],
       # TODO(davidben): Fix size_t truncations in BoringSSL.
       # https://crbug.com/429039
