--- third_party/WebKit/Source/bindings/derived_sources.gyp.orig	2013-07-30 22:29:53.610647573 +0000
+++ third_party/WebKit/Source/bindings/derived_sources.gyp	2013-07-30 22:30:05.150721649 +0000
@@ -262,8 +262,6 @@
         'msvs_external_rule': 1,
         'inputs': [
           'scripts/idl_compiler.py',
-          '<(DEPTH)/third_party/ply/lex.py',
-          '<(DEPTH)/third_party/ply/yacc.py',
           '<(DEPTH)/tools/idl_parser/idl_lexer.py',
           '<(DEPTH)/tools/idl_parser/idl_node.py',
           '<(DEPTH)/tools/idl_parser/idl_parser.py',
