--- third_party/WebKit/Source/bindings/generated_bindings.gyp.orig	2014-02-20 03:49:21.821527295 +0000
+++ third_party/WebKit/Source/bindings/generated_bindings.gyp	2014-02-20 03:49:34.401607238 +0000
@@ -110,9 +110,6 @@
       'scripts/unstable/idl_compiler.py',
       '<(DEPTH)/third_party/ply/lex.py',
       '<(DEPTH)/third_party/ply/yacc.py',
-      # jinja2/__init__.py contains version string, so sufficient for package
-      '<(DEPTH)/third_party/jinja2/__init__.py',
-      '<(DEPTH)/third_party/markupsafe/__init__.py',  # jinja2 dep
       '<(DEPTH)/tools/idl_parser/idl_lexer.py',
       '<(DEPTH)/tools/idl_parser/idl_node.py',
       '<(DEPTH)/tools/idl_parser/idl_parser.py',
--- third_party/WebKit/Source/build/scripts/scripts.gypi.orig	2014-02-20 03:49:59.351765796 +0000
+++ third_party/WebKit/Source/build/scripts/scripts.gypi	2014-02-20 03:50:08.111821470 +0000
@@ -1,10 +1,6 @@
 {
     'variables': {
         'scripts_for_in_files': [
-            # jinja2/__init__.py contains version string, so sufficient as
-            # dependency for whole jinja2 package
-            '<(DEPTH)/third_party/jinja2/__init__.py',
-            '<(DEPTH)/third_party/markupsafe/__init__.py',  # jinja2 dep
             'hasher.py',
             'in_file.py',
             'in_generator.py',
