--- media/cast/logging/logging.gyp.orig	2014-03-18 22:32:33.802818765 +0000
+++ media/cast/logging/logging.gyp	2014-03-18 22:33:14.683093733 +0000
@@ -40,6 +40,7 @@
         'cast_common_logging',
         'cast_logging_proto_lib',
         '<(DEPTH)/base/base.gyp:base',
+        '<(DEPTH)/third_party/zlib/zlib.gyp:zlib',
       ],
       'export_dependent_settings': [
         'cast_logging_proto_lib',
