--- third_party/speex/speex.h.orig	2012-10-01 17:40:03.000000000 +0200
+++ third_party/speex/speex.h	2012-10-01 17:41:30.000000000 +0200
@@ -11,8 +11,12 @@
 
 #if defined(USE_SYSTEM_SPEEX)
 #include <speex/speex.h>
+#include <speex/speex_callbacks.h>
+#include <speex/speex_stereo.h>
 #else
 #include "third_party/speex/include/speex/speex.h"
+#include "third_party/speex/include/speex/speex_callbacks.h"
+#include "third_party/speex/include/speex/speex_stereo.h"
 #endif
 
 #endif  // THIRD_PARTY_SPEEX_SPEEX_H_
--- remoting/codec/audio_decoder_speex.cc.orig	2012-10-01 17:40:21.000000000 +0200
+++ remoting/codec/audio_decoder_speex.cc	2012-10-01 17:40:44.000000000 +0200
@@ -12,8 +12,6 @@
 #include "base/stl_util.h"
 #include "remoting/proto/audio.pb.h"
 #include "third_party/speex/speex.h"
-#include "third_party/speex/include/speex/speex_callbacks.h"
-#include "third_party/speex/include/speex/speex_stereo.h"
 
 namespace remoting {
 
--- remoting/codec/audio_encoder_speex.cc.orig	2012-10-01 17:56:56.000000000 +0200
+++ remoting/codec/audio_encoder_speex.cc	2012-10-01 17:57:08.000000000 +0200
@@ -12,7 +12,6 @@
 #include "base/stl_util.h"
 #include "remoting/proto/audio.pb.h"
 #include "third_party/speex/speex.h"
-#include "third_party/speex/include/speex/speex_stereo.h"
 
 namespace {
 // A quality of 8 in wide band mode corresponds to 27,800 bits per second.
