Index: build/linux/system.gyp
===================================================================
--- build/linux/system.gyp	(revision 158796)
+++ build/linux/system.gyp	(working copy)
@@ -112,7 +112,28 @@
           ],
         },
       ],  # targets
-    }]  # chromeos==0
+    }, {  # chromeos==1
+      'targets': [
+        {
+          # TODO(satorux): Remove this once dbus-glib clients are gone.
+          'target_name': 'dbus-glib',
+          'type': 'none',
+          'direct_dependent_settings': {
+            'cflags': [
+              '<!@(<(pkg-config) --cflags dbus-glib-1)',
+            ],
+          },
+          'link_settings': {
+            'ldflags': [
+              '<!@(<(pkg-config) --libs-only-L --libs-only-other dbus-glib-1)',
+            ],
+            'libraries': [
+              '<!@(<(pkg-config) --libs-only-l dbus-glib-1)',
+            ],
+          },
+        },
+      ],
+    }]
   ],  # conditions
   'targets': [
     {
@@ -477,24 +498,6 @@
       },
     },
     {
-      # TODO(satorux): Remove this once dbus-glib clients are gone.
-      'target_name': 'dbus-glib',
-      'type': 'none',
-      'direct_dependent_settings': {
-        'cflags': [
-          '<!@(<(pkg-config) --cflags dbus-glib-1)',
-        ],
-      },
-      'link_settings': {
-        'ldflags': [
-          '<!@(<(pkg-config) --libs-only-L --libs-only-other dbus-glib-1)',
-        ],
-        'libraries': [
-          '<!@(<(pkg-config) --libs-only-l dbus-glib-1)',
-        ],
-      },
-    },
-    {
       'target_name': 'glib',
       'type': 'none',
       'toolsets': ['host', 'target'],
