--- skia/skia.gyp.orig	2012-10-29 11:38:14.026722118 -0500
+++ skia/skia.gyp	2012-10-29 11:39:11.856020934 -0500
@@ -747,9 +747,11 @@
         [ 'target_arch == "arm" and armv7 != 1', {
           'sources': [
             '../third_party/skia/src/opts/SkBlitRow_opts_none.cpp',
+            '../third_party/skia/src/opts/SkUtils_opts_none.cpp',
           ],
           'sources!': [
             '../third_party/skia/src/opts/SkBlitRow_opts_arm.cpp',
+            '../third_party/skia/src/opts/opts_check_arm.cpp',
           ],
         }],
       ],
