diff -Naur gnome-control-center-3.8.3_orig/configure.ac gnome-control-center-3.8.3/configure.ac
--- gnome-control-center-3.8.3_orig/configure.ac	2013-06-08 00:11:56.000000000 +0200
+++ gnome-control-center-3.8.3/configure.ac	2013-10-04 12:55:50.470708998 +0200
@@ -128,6 +128,7 @@
 PKG_CHECK_MODULES(BACKGROUND_PANEL, $COMMON_MODULES libxml-2.0 gnome-desktop-3.0
                   gdk-pixbuf-2.0 >= $GDKPIXBUF_REQUIRED_VERSION)
 PKG_CHECK_MODULES(DATETIME_PANEL, $COMMON_MODULES
+                  timezonemap
                   gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION
                   polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION
                   gdk-pixbuf-2.0 >= $GDKPIXBUF_REQUIRED_VERSION)
diff -Naur gnome-control-center-3.8.3_orig/panels/datetime/Makefile.am gnome-control-center-3.8.3/panels/datetime/Makefile.am
--- gnome-control-center-3.8.3_orig/panels/datetime/Makefile.am	2013-01-25 15:42:18.000000000 +0100
+++ gnome-control-center-3.8.3/panels/datetime/Makefile.am	2013-10-04 13:20:25.250120897 +0200
@@ -28,7 +28,7 @@
 
 noinst_PROGRAMS = test-timezone-gfx test-endianess test-timezone
 
-test_timezone_SOURCES = test-timezone.c cc-timezone-map.h cc-timezone-map.c tz.c tz.h
+test_timezone_SOURCES = test-timezone.c tz.c tz.h
 test_timezone_LDADD = $(DATETIME_PANEL_LIBS)
 test_timezone_CFLAGS = $(DATETIME_PANEL_CFLAGS)
 
@@ -84,11 +84,8 @@
 	$(BUILT_SOURCES)	\
 	cc-datetime-panel.c	\
 	cc-datetime-panel.h	\
-	cc-timezone-map.c	\
-	cc-timezone-map.h	\
 	date-endian.c		\
 	date-endian.h		\
-	tz.c tz.h		\
 	$(NULL)
 
 libdate_time_la_LIBADD = $(PANEL_LIBS) $(DATETIME_PANEL_LIBS)
