--- gnome-control-center-3.8.1.5-orig/configure.ac	2013-05-06 18:32:26.000000000 +0200
+++ gnome-control-center-3.8.1.5-dev/configure.ac	2013-05-11 22:57:47.955153458 +0200
@@ -200,8 +200,25 @@
 fi
 
 # Check for libmm-glib for ModemManager1 support
-PKG_CHECK_MODULES(MM_GLIB, mm-glib,
-                  [have_libmm_glib=yes], have_libmm_glib=no)
+build_modemmanager=false
+AC_ARG_ENABLE(modemmanager,
+	AC_HELP_STRING([--disable-modemmanager],
+		[disable modemmanager support]),
+	[case "${enableval}" in
+		yes) WANT_MODEMMANAGER=yes ;;
+		no)  WANT_MODEMMANAGER=no ;;
+		*) AC_MSG_ERROR(bad value ${enableval} for --disable-modemmanager) ;;
+	esac],
+	[WANT_MODEMMANAGER=no]) dnl Default value
+
+if test "x$WANT_MODEMMANAGER" = xyes ; then
+	PKG_CHECK_MODULES(MM_GLIB, mm-glib,
+                  [have_libmm_glib=yes],
+		  AC_MSG_ERROR(No ModemManager1 support (libmm-glib) not found))
+else
+	have_libmm_glib=no
+fi
+
 if test "x$have_libmm_glib" = xno ; then
   AC_MSG_WARN(*** Network panel will not be built with ModemManager1 support (libmm-glib not found) ***)
 else
