diff -Naur gnome-control-center-3.8.3_orig/panels/info/cc-info-panel.c gnome-control-center-3.8.3/panels/info/cc-info-panel.c
--- gnome-control-center-3.8.3_orig/panels/info/cc-info-panel.c	2013-06-05 11:27:56.000000000 +0200
+++ gnome-control-center-3.8.3/panels/info/cc-info-panel.c	2013-09-27 15:27:28.466276594 +0200
@@ -310,6 +310,7 @@
     None
   };
   int nconfigs;
+  int major, minor;
   Window window;
   GLXFBConfig *config;
   GLXWindow glxwin;
@@ -322,6 +323,7 @@
 
   display = GDK_DISPLAY_XDISPLAY (gdk_display_get_default ());
 
+  glXQueryVersion (display, &major, &minor);
   config = glXChooseFBConfig (display, DefaultScreen (display),
                               attributes, &nconfigs);
   if (config == NULL) {
