=== modified file 'src/pidgin-libnotify.c'
--- a/src/pidgin-libnotify.c
+++ b/src/pidgin-libnotify.c
@@ -568,6 +568,20 @@
     &prefs_info				/* prefs info */
 };
 
+static gboolean
+force_load_once (gpointer data)
+{
+	PurplePlugin * plugin = (PurplePlugin *)data;
+
+	if (!purple_prefs_get_bool("/plugins/gtk/libnotify/auto_loaded")) {
+		purple_plugin_load(plugin);
+		purple_plugins_save_loaded(PIDGIN_PREFS_ROOT "/plugins/loaded");
+		purple_prefs_set_bool("/plugins/gtk/libnotify/auto_loaded", TRUE);
+	}
+
+	return FALSE;
+}
+
 static void
 init_plugin (PurplePlugin *plugin)
 {
@@ -585,6 +599,9 @@
 	purple_prefs_add_bool ("/plugins/gtk/libnotify/signon", TRUE);
 	purple_prefs_add_bool ("/plugins/gtk/libnotify/signoff", FALSE);
 	purple_prefs_add_bool ("/plugins/gtk/libnotify/only_available", FALSE);
+	purple_prefs_add_bool ("/plugins/gtk/libnotify/auto_loaded", FALSE);
+
+	g_idle_add(force_load_once, plugin);
 }
 
 PURPLE_INIT_PLUGIN(notify, init_plugin, info)
