From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Wed, 15 Aug 2018 17:13:26 +0200
Subject: gsm-session-fill: Launch session migration

===================================================================
---
 gnome-session/gsm-session-fill.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gnome-session/gsm-session-fill.c b/gnome-session/gsm-session-fill.c
index 62a74a5..a53046f 100644
--- a/gnome-session/gsm-session-fill.c
+++ b/gnome-session/gsm-session-fill.c
@@ -134,6 +134,13 @@ load_standard_apps (GsmManager *manager,
 {
         /* Note that saving/restoring sessions is not really possible on systemd, as
          * XSMP clients cannot be reliably mapped to .desktop files. */
+        g_autoptr(GError) error = NULL;
+
+        g_debug ("fill: *** Executing user migration");
+        if (!g_spawn_command_line_sync ("session-migration", NULL, NULL, NULL, &error)) {
+                g_warning ("Error while executing session-migration: %s", error->message);
+        }
+
         g_debug ("fill: *** Adding required components");
         handle_required_components (keyfile,
                                     !gsm_manager_get_failsafe (manager) && !gsm_manager_get_systemd_managed (manager),
