From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Wed, 15 Aug 2018 17:13:26 +0200
Subject: Remove session saving from gui

===================================================================
---
 gnome-session/gsm-session-fill.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gnome-session/gsm-session-fill.c b/gnome-session/gsm-session-fill.c
index bdd9208..62a74a5 100644
--- a/gnome-session/gsm-session-fill.c
+++ b/gnome-session/gsm-session-fill.c
@@ -146,7 +146,8 @@ load_standard_apps (GsmManager *manager,
 
                 autostart_dirs = gsm_util_get_autostart_dirs ();
 
-                if (!gsm_manager_get_systemd_managed (manager))
+                if (!gsm_manager_get_systemd_managed (manager) &&
+                    g_getenv ("GNOME_SESSION_SAVE") != NULL)
                         maybe_load_saved_session_apps (manager);
 
                 for (i = 0; autostart_dirs[i]; i++) {
