diff -urN NetworkManager-1.2.2/libnm/Makefile.in NetworkManager-1.2.2-patched/libnm/Makefile.in
--- NetworkManager-1.2.2/libnm/Makefile.in	2016-05-11 20:54:18.000000000 +0930
+++ NetworkManager-1.2.2-patched/libnm/Makefile.in	2016-07-11 07:50:54.372966460 +0930
@@ -913,7 +913,7 @@
 	}
 
 libnm.la: $(libnm_la_OBJECTS) $(libnm_la_DEPENDENCIES) $(EXTRA_libnm_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libnm_la_LINK) -rpath $(libdir) $(libnm_la_OBJECTS) $(libnm_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libnm_la_LINK) -rpath $(libdir) $(libnm_la_OBJECTS) $(libnm_la_LIBADD) $(LIBS) $(LIBDL)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
