=== modified file 'src/client/mir_connection.cpp'
--- src/client/mir_connection.cpp	2014-01-31 03:46:36 +0000
+++ src/client/mir_connection.cpp	2014-02-17 09:15:15 +0000
@@ -458,6 +458,7 @@
             display_request->set_output_id(output.output_id);
             display_request->set_used(output.used);
             display_request->set_current_mode(output.current_mode);
+            display_request->set_current_format(output.current_format);
             display_request->set_position_x(output.position_x);
             display_request->set_position_y(output.position_y);
             display_request->set_power_mode(output.power_mode);

=== modified file 'tests/unit-tests/client/test_mir_connection.cpp'
--- tests/unit-tests/client/test_mir_connection.cpp	2014-01-14 04:10:15 +0000
+++ tests/unit-tests/client/test_mir_connection.cpp	2014-02-17 09:32:44 +0000
@@ -455,6 +455,7 @@
     MirDisplayOutput output;
     output.output_id = 0;
     output.current_mode = 0;
+    output.current_format = mir_pixel_format_xbgr_8888;
     output.used = 0;
     output.position_x = 4;
     output.position_y = 6;
@@ -466,10 +467,17 @@
         ASSERT_NE(nullptr, config);
         ASSERT_EQ(1, config->display_output_size());
         auto const& disp1 = config->display_output(0);
+        EXPECT_TRUE(disp1.has_output_id());
         EXPECT_EQ(output.output_id, disp1.output_id());
+        EXPECT_TRUE(disp1.has_used());
         EXPECT_EQ(output.used, disp1.used());
+        EXPECT_TRUE(disp1.has_current_mode());
         EXPECT_EQ(output.current_mode, disp1.current_mode());
+        EXPECT_TRUE(disp1.has_current_format());
+        EXPECT_EQ(output.current_format, disp1.current_format());
+        EXPECT_TRUE(disp1.has_position_x());
         EXPECT_EQ(output.position_x, disp1.position_x());
+        EXPECT_TRUE(disp1.has_position_y());
         EXPECT_EQ(output.position_y, disp1.position_y());
     };
 

