diff -urN indicator-session-12.10.5+13.10.20130812/src/service.c indicator-session-12.10.5+13.10.20130812-patched/src/service.c
--- indicator-session-12.10.5+13.10.20130812/src/service.c	2013-08-13 04:09:24.000000000 +0930
+++ indicator-session-12.10.5+13.10.20130812-patched/src/service.c	2013-08-18 10:40:24.374592390 +0930
@@ -286,7 +286,7 @@
 
   menu = g_menu_new ();
   g_menu_append (menu, _("About This Computer"), "indicator.about");
-  g_menu_append (menu, _("Ubuntu Help"), "indicator.help");
+//  g_menu_append (menu, _("Ubuntu Help"), "indicator.help");
   return G_MENU_MODEL (menu);
 }
 
