diff -urN indicator-session-12.10.5+13.10.20130812/cmake/UseGSettings.cmake indicator-session-12.10.5+13.10.20130812-patched/cmake/UseGSettings.cmake
--- indicator-session-12.10.5+13.10.20130812/cmake/UseGSettings.cmake	2013-08-13 04:09:24.000000000 +0930
+++ indicator-session-12.10.5+13.10.20130812-patched/cmake/UseGSettings.cmake	2013-08-17 12:52:22.554072745 +0930
@@ -10,14 +10,14 @@
     execute_process (COMMAND ${_glib_compile_schemas} --dry-run --schema-file=${SCHEMA_NAME} ERROR_VARIABLE _schemas_invalid OUTPUT_STRIP_TRAILING_WHITESPACE)
 
     if (_schemas_invalid)
-      message (SEND_ERROR "Schema validation error: ${_schemas_invalid}")
+      message (WARNING "Schema validation error: ${_schemas_invalid}")
     endif (_schemas_invalid)
 
     # Actually install and recomple schemas
     message (STATUS "${GSETTINGS_DIR} is the GSettings install dir")
     install (FILES ${SCHEMA_NAME} DESTINATION ${GSETTINGS_DIR} OPTIONAL)
 
-    install (CODE "message (STATUS \"Compiling GSettings schemas\")")
-    install (CODE "execute_process (COMMAND ${_glib_compile_schemas} ${GSETTINGS_DIR})")
+#    install (CODE "message (STATUS \"Compiling GSettings schemas\")")
+#    install (CODE "execute_process (COMMAND ${_glib_compile_schemas} ${GSETTINGS_DIR})")
 endmacro()
 
diff -urN indicator-session-12.10.5+13.10.20130812/data/CMakeLists.txt indicator-session-12.10.5+13.10.20130812-patched/data/CMakeLists.txt
--- indicator-session-12.10.5+13.10.20130812/data/CMakeLists.txt	2013-08-17 12:51:44.254070891 +0930
+++ indicator-session-12.10.5+13.10.20130812-patched/data/CMakeLists.txt	2013-08-17 12:52:02.922071795 +0930
@@ -69,6 +69,6 @@
          DESTINATION "${ICON_DIR}"
          FILES_MATCHING PATTERN "*.png" PATTERN "*.svg")
 
-install(CODE "execute_process (COMMAND gtk-update-icon-cache -t -f ${ICON_DIR})"
-        CODE "message (STATUS \"Updating icon cache\")")
+#install(CODE "execute_process (COMMAND gtk-update-icon-cache -t -f ${ICON_DIR})"
+#        CODE "message (STATUS \"Updating icon cache\")")
 
