diff -Nur a/src/migration.c b/src/migration.c
--- a/src/migration.c	2016-05-23 10:28:00.000000000 +0200
+++ b/src/migration.c	2017-08-18 03:11:43.420310711 +0200
@@ -331,6 +331,9 @@
   migrate_string_list (client, GCONF_GLOBAL_PREFIX, "active_encodings",
                        settings, TERMINAL_SETTING_ENCODINGS_KEY);
 
+  if (verbose)
+      g_printerr("Migrating global\n");
+
   g_object_unref (client);
 
   return TRUE;
@@ -378,6 +381,8 @@
 {
   GSettings *settings;
   char *child_name, *path;
+  gchar *background_type = NULL, *bg_type_key, *bg_darkness_key;
+  gdouble background_darkness;
   gs_free char *name;
   gboolean is_default;
 
@@ -394,6 +399,31 @@
 
   path = gconf_concat_dir_and_key (GCONF_PROFILES_PREFIX, gconf_id);
 
+  bg_type_key = gconf_concat_dir_and_key (path, KEY_BACKGROUND_TYPE);
+  background_type = gconf_client_get_string (client, bg_type_key, NULL);
+  g_free (bg_type_key);
+  if (background_type)
+      g_settings_set_boolean (settings, TERMINAL_PROFILE_USE_TRANSPARENT_BACKGROUND,
+                              g_strcmp0(background_type, "transparent") == 0);
+  g_free (background_type);
+
+  // If we're migrating, set the 'use theme transparency' key to false, this
+  // will be copied from the old profile into transparency.
+  g_settings_set_boolean (settings, TERMINAL_PROFILE_USE_THEME_TRANSPARENCY,
+                          FALSE);
+
+  bg_darkness_key = gconf_concat_dir_and_key (path, KEY_BACKGROUND_DARKNESS);
+  background_darkness = gconf_client_get_float (client,
+                                                bg_darkness_key, NULL);
+
+  g_free (bg_darkness_key);
+  if (background_darkness) {
+      // Need to invert, the key has changed sense
+      g_settings_set_int (settings,
+                          TERMINAL_PROFILE_BACKGROUND_TRANSPARENCY_PERCENT,
+                          1 - (int) (background_darkness * 100));
+  }
+
   migrate_string (client, path, KEY_VISIBLE_NAME,
                   settings, TERMINAL_PROFILE_VISIBLE_NAME_KEY);
 
@@ -481,7 +511,7 @@
   TerminalSettingsList *list;
   GConfClient *client;
   GConfValue *value, *dvalue;
-  GSList *l;
+  GSList *l, *def = NULL;
   const char *default_profile;
 
   client = gconf_client_get_default ();
@@ -491,15 +521,24 @@
       dvalue->type == GCONF_VALUE_STRING)
     default_profile = gconf_value_get_string (dvalue);
   else
-    default_profile = NULL;
+    default_profile = "Default";
 
   list = terminal_profiles_list_new ();
 
   value = gconf_client_get (client, GCONF_GLOBAL_PREFIX "/profile_list", NULL);
+  if (value == NULL) {
+      value = gconf_value_new (GCONF_VALUE_LIST);
+      gconf_value_set_list_type (value, GCONF_VALUE_STRING);
+      def = g_slist_append(def, gconf_value_new_from_string(GCONF_VALUE_STRING,
+                                                            "Default", NULL));
+      gconf_value_set_list_nocopy (value, def);
+  }
   if (value != NULL &&
       value->type == GCONF_VALUE_LIST &&
       gconf_value_get_list_type (value) == GCONF_VALUE_STRING) {
     for (l = gconf_value_get_list (value); l != NULL; l = l->next) {
+      if (verbose)
+          g_printerr ("Migrating %s\n", gconf_value_get_string (l->data));
       migrate_profile (list, client,
                        gconf_value_get_string (l->data),
                        default_profile);
diff -Nur a/src/org.gnome.Terminal.gschema.xml b/src/org.gnome.Terminal.gschema.xml
--- a/src/org.gnome.Terminal.gschema.xml	2017-08-18 03:09:24.706859584 +0200
+++ b/src/org.gnome.Terminal.gschema.xml	2017-08-18 03:11:43.415310658 +0200
@@ -374,9 +374,13 @@
       <default>false</default>
       <summary>Whether to use a transparent background</summary>
     </key>
+    <key name="use-theme-transparency" type="b">
+      <default>true</default>
+      <summary>Whether to use the value of TerminalScreen-background-darkness,
+          if available, from the theme for the transparency value.</summary>
+    </key>
     <key name="background-transparency-percent" type="i">
       <default>50</default>
-      <range min="0" max="100"/>
       <summary>Adjust the amount of transparency</summary>
       <description>A value between 0 and 100, where 0 is opaque and 100 is fully transparent.</description>
     </key>
diff -Nur a/src/profile-editor.c b/src/profile-editor.c
--- a/src/profile-editor.c	2017-08-18 03:09:24.706859584 +0200
+++ b/src/profile-editor.c	2017-08-18 03:11:43.416310668 +0200
@@ -35,6 +35,7 @@
 #include "terminal-util.h"
 #include "terminal-profiles-list.h"
 #include "terminal-libgsystem.h"
+#include "terminal-window.h"
 
 typedef struct _TerminalColorScheme TerminalColorScheme;
 
@@ -817,6 +818,7 @@
   GtkWidget *editor, *w;
   gs_free char *uuid = NULL;
   guint i;
+  gfloat style_darkness;
 
   editor = g_object_get_data (G_OBJECT (profile), "editor-window");
   if (editor)
@@ -824,7 +826,7 @@
       terminal_util_dialog_focus_widget (editor, widget_name);
 
       gtk_window_set_transient_for (GTK_WINDOW (editor),
-                                    GTK_WINDOW (transient_parent));
+                                    NULL);
       gtk_window_present (GTK_WINDOW (editor));
       return;
     }
@@ -905,6 +907,15 @@
                         profile);
     }
 
+  gtk_widget_style_get (GTK_WIDGET (
+              terminal_window_get_active (TERMINAL_WINDOW (transient_parent))),
+                        "background-darkness", &style_darkness,
+                        NULL);
+
+  gtk_widget_set_visible (gtk_builder_get_object (
+              builder,
+              "use-theme-transparency-checkbutton"), style_darkness >= 0);
+
   profile_palette_notify_colorpickers_cb (profile, TERMINAL_PROFILE_PALETTE_KEY, editor);
   g_signal_connect (profile, "changed::" TERMINAL_PROFILE_PALETTE_KEY,
                     G_CALLBACK (profile_palette_notify_colorpickers_cb),
@@ -1224,14 +1235,24 @@
   g_settings_bind (profile, TERMINAL_PROFILE_BACKGROUND_TRANSPARENCY_PERCENT,
                    gtk_builder_get_object (builder, "background-transparent-adjustment"),
                    "value", G_SETTINGS_BIND_GET | G_SETTINGS_BIND_SET);
+  g_settings_bind (profile, TERMINAL_PROFILE_USE_THEME_TRANSPARENCY,
+                   gtk_builder_get_object (builder, "use-theme-transparency-checkbutton"),
+                   "active", G_SETTINGS_BIND_GET | G_SETTINGS_BIND_SET);
 
-  /* Finished! */
+  if (style_darkness >= 0)
+      g_settings_bind (profile, TERMINAL_PROFILE_USE_THEME_TRANSPARENCY,
+              gtk_builder_get_object (builder, "use-transparent-background-box"),
+              "sensitive",
+              G_SETTINGS_BIND_GET |
+              G_SETTINGS_BIND_INVERT_BOOLEAN |
+              G_SETTINGS_BIND_NO_SENSITIVITY);
 
+  /* Finished! */
   terminal_util_bind_mnemonic_label_sensitivity (editor);
 
   terminal_util_dialog_focus_widget (editor, widget_name);
 
   gtk_window_set_transient_for (GTK_WINDOW (editor),
-                                GTK_WINDOW (transient_parent));
+                                NULL);
   gtk_window_present (GTK_WINDOW (editor));
 }
diff -Nur a/src/profile-preferences.ui b/src/profile-preferences.ui
--- a/src/profile-preferences.ui	2017-08-18 03:09:24.707859594 +0200
+++ b/src/profile-preferences.ui	2017-08-18 03:11:43.417310679 +0200
@@ -23,11 +23,6 @@
     <property name="step_increment">1</property>
     <property name="page_increment">100</property>
   </object>
-  <object class="GtkAdjustment" id="background-transparent-adjustment">
-    <property name="upper">100</property>
-    <property name="step_increment">1</property>
-    <property name="page_increment">10</property>
-  </object>
   <object class="GtkListStore" id="cjk-ambiguous-width-model">
     <columns>
       <!-- column-name gchararray -->
@@ -46,6 +41,11 @@
       </row>
     </data>
   </object>
+  <object class="GtkAdjustment" id="background-transparent-adjustment">
+    <property name="upper">100</property>
+    <property name="step_increment">1</property>
+    <property name="page_increment">10</property>
+  </object>
   <object class="GtkListStore" id="model1">
     <columns>
       <!-- column-name gchararray -->
@@ -1049,7 +1049,7 @@
                                 <property name="spacing">12</property>
                                 <child>
                                   <object class="GtkCheckButton" id="use-transparent-background">
-                                    <property name="label" translatable="yes">Transparent background</property>
+                                    <property name="label" translatable="yes">Use t_ransparent background</property>
                                     <property name="visible">True</property>
                                     <property name="can_focus">True</property>
                                     <property name="receives_default">False</property>
@@ -1128,6 +1128,35 @@
                                 <property name="position">2</property>
                               </packing>
                             </child>
+                            <child>
+                              <object class="GtkBox" id="use-theme-transparency-box">
+                                <property name="visible">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="orientation">horizontal</property>
+                                <property name="spacing">12</property>
+                                <child>
+                                  <object class="GtkCheckButton" id="use-theme-transparency-checkbutton">
+                                    <property name="label" translatable="yes">Use transparency from system theme</property>
+                                    <property name="visible">True</property>
+                                    <property name="can_focus">True</property>
+                                    <property name="receives_default">False</property>
+                                    <property name="use_underline">True</property>
+                                    <property name="xalign">0</property>
+                                    <property name="draw_indicator">True</property>
+                                  </object>
+                                  <packing>
+                                    <property name="expand">False</property>
+                                    <property name="fill">False</property>
+                                    <property name="position">0</property>
+                                  </packing>
+                                </child>
+                              </object>
+                              <packing>
+                                <property name="expand">True</property>
+                                <property name="fill">True</property>
+                                <property name="position">3</property>
+                              </packing>
+                            </child>
                           </object>
                         </child>
                       </object>
diff -Nur a/src/terminal-app.c b/src/terminal-app.c
--- a/src/terminal-app.c	2017-03-20 22:02:06.000000000 +0100
+++ b/src/terminal-app.c	2017-08-18 03:11:43.421310721 +0200
@@ -237,7 +237,7 @@
   if (profile == NULL)
     return;
 
-  terminal_profile_edit (profile, NULL, "profile-name-entry");
+  terminal_profile_edit (profile, transient_parent, "profile-name-entry");
 }
 
 void
@@ -339,8 +339,11 @@
                          gpointer       user_data)
 {
   TerminalApp *app = user_data;
+  GtkWindow *window;
+
+  window = gtk_application_get_active_window (GTK_APPLICATION (app));
 
-  terminal_app_edit_preferences (app, NULL);
+  terminal_app_edit_preferences (app, window);
 }
 
 static void
@@ -675,21 +678,21 @@
                            GtkWindow       *transient_parent,
                            const char      *widget_name)
 {
-  terminal_profile_edit (profile, NULL, widget_name);
+  terminal_profile_edit (profile, transient_parent, widget_name);
 }
 
 void
 terminal_app_edit_preferences (TerminalApp     *app,
                                GtkWindow       *transient_parent)
 {
-  terminal_prefs_show_preferences (NULL, "general");
+  terminal_prefs_show_preferences (transient_parent, "general");
 }
 
 void
 terminal_app_edit_encodings (TerminalApp     *app,
                              GtkWindow       *transient_parent)
 {
-  terminal_prefs_show_preferences (NULL, "encodings");
+  terminal_prefs_show_preferences (transient_parent, "encodings");
 }
 
 /**
diff -Nur a/src/terminal-prefs.c b/src/terminal-prefs.c
--- a/src/terminal-prefs.c	2016-11-27 20:10:27.000000000 +0100
+++ b/src/terminal-prefs.c	2017-08-18 03:11:43.421310721 +0200
@@ -300,7 +300,7 @@
   if (selected_profile == NULL)
     return;
 
-  terminal_app_edit_profile (terminal_app_get (), selected_profile, NULL, NULL);
+  terminal_app_edit_profile (terminal_app_get (), selected_profile, GTK_WINDOW (data->parent), NULL);
 }
 
 static GtkTreeView *
@@ -425,7 +425,7 @@
   if (selected_profile == NULL)
     return;
 
-  terminal_app_edit_profile (terminal_app_get (), selected_profile, NULL, NULL);
+  terminal_app_edit_profile (terminal_app_get (), selected_profile, GTK_WINDOW (data->parent), NULL);
 }
 
 static void
diff -Nur a/src/terminal-schemas.h b/src/terminal-schemas.h
--- a/src/terminal-schemas.h	2017-08-18 03:09:24.707859594 +0200
+++ b/src/terminal-schemas.h	2017-08-18 03:11:43.418310690 +0200
@@ -70,6 +70,7 @@
 #define TERMINAL_PROFILE_WORD_CHAR_EXCEPTIONS_KEY       "word-char-exceptions"
 
 #define TERMINAL_PROFILE_USE_TRANSPARENT_BACKGROUND      "use-transparent-background"
+#define TERMINAL_PROFILE_USE_THEME_TRANSPARENCY          "use-theme-transparency"
 #define TERMINAL_PROFILE_BACKGROUND_TRANSPARENCY_PERCENT "background-transparency-percent"
 
 #define TERMINAL_SETTING_CONFIRM_CLOSE_KEY              "confirm-close"
diff -Nur a/src/terminal-screen.c b/src/terminal-screen.c
--- a/src/terminal-screen.c	2017-08-18 03:09:24.708859605 +0200
+++ b/src/terminal-screen.c	2017-08-18 03:11:43.418310690 +0200
@@ -563,6 +563,11 @@
 
   g_type_class_add_private (object_class, sizeof (TerminalScreenPrivate));
 
+  gtk_widget_class_install_style_property (widget_class,
+     g_param_spec_float ("background-darkness", NULL, NULL, -1, 1, -1,
+                         G_PARAM_READABLE | G_PARAM_STATIC_NAME | G_PARAM_STATIC_NICK | G_PARAM_STATIC_BLURB));
+
+
   n_url_regexes = G_N_ELEMENTS (url_regex_patterns);
   precompile_regexes (url_regex_patterns, n_url_regexes, &url_regexes, &url_regex_flavors);
   n_extra_regexes = G_N_ELEMENTS (extra_regex_patterns);
@@ -829,7 +834,8 @@
       prop_name == I_(TERMINAL_PROFILE_HIGHLIGHT_FOREGROUND_COLOR_KEY) ||
       prop_name == I_(TERMINAL_PROFILE_PALETTE_KEY) ||
       prop_name == I_(TERMINAL_PROFILE_USE_TRANSPARENT_BACKGROUND) ||
-      prop_name == I_(TERMINAL_PROFILE_BACKGROUND_TRANSPARENCY_PERCENT))
+      prop_name == I_(TERMINAL_PROFILE_BACKGROUND_TRANSPARENCY_PERCENT) ||
+      prop_name == I_(TERMINAL_PROFILE_USE_THEME_TRANSPARENCY))
     update_color_scheme (screen);
 
   if (!prop_name || prop_name == I_(TERMINAL_PROFILE_AUDIBLE_BELL_KEY))
@@ -899,9 +905,10 @@
   GdkRGBA *cursor_bgp = NULL, *cursor_fgp = NULL;
   GdkRGBA *highlight_bgp = NULL, *highlight_fgp = NULL;
   GtkStyleContext *context;
-  GtkWidget *toplevel;
-  gboolean transparent;
   gboolean use_theme_colors;
+  GtkWidget *toplevel;
+  gboolean transparent, theme_transparent;
+  gfloat style_darkness;
 
   context = gtk_widget_get_style_context (widget);
   gtk_style_context_get_color (context, gtk_style_context_get_state (context), &theme_fg);
@@ -942,9 +949,18 @@
     }
 
   colors = terminal_g_settings_get_rgba_palette (priv->profile, TERMINAL_PROFILE_PALETTE_KEY, &n_colors);
-
+  theme_transparent = g_settings_get_boolean (profile, TERMINAL_PROFILE_USE_THEME_TRANSPARENCY);
   transparent = g_settings_get_boolean (profile, TERMINAL_PROFILE_USE_TRANSPARENT_BACKGROUND);
-  if (transparent)
+
+  gtk_widget_style_get (GTK_WIDGET (screen),
+                        "background-darkness", &style_darkness,
+                        NULL);
+
+  if (theme_transparent && style_darkness >= 0)
+    {
+      bg.alpha = style_darkness;
+    }
+  else if (transparent)
     {
       gint transparency_percent;
 
@@ -952,7 +968,11 @@
       bg.alpha = (100 - transparency_percent) / 100.0;
     }
   else
-    bg.alpha = 1.0;
+      bg.alpha = 1.0;
+
+  /* If this gets out of range, let's not crash */
+  if (bg.alpha < 0.0 || bg.alpha > 1.0)
+      bg.alpha = 1.0;
 
   vte_terminal_set_colors (VTE_TERMINAL (screen), &fg, &bg,
                            colors, n_colors);
