diff -Naur deja-dup-29.1_orig/deja-dup/widgets/WidgetUtils.vala deja-dup-29.1/deja-dup/widgets/WidgetUtils.vala
--- deja-dup-29.1_orig/deja-dup/widgets/WidgetUtils.vala	2013-10-14 16:17:59.000000000 +0200
+++ deja-dup-29.1/deja-dup/widgets/WidgetUtils.vala	2013-12-31 14:53:12.510214670 +0100
@@ -46,6 +46,12 @@
 }
 
 protected ShellEnv shell = ShellEnv.NONE;
+
+#if !VALA_0_24
+  [CCode (cheader_filename = "libnotify/notify.h", cname = "notify_get_server_caps")]
+  extern static GLib.List<string> notify_get_server_caps ();
+#endif
+
 public ShellEnv get_shell()
 {
   if (shell == ShellEnv.NONE) {
@@ -59,14 +65,16 @@
       // Use Legacy unless we detect a different shell.
       shell = ShellEnv.LEGACY;
       // Next check for Shell by notification capabilities
-      unowned List<string> caps = Notify.get_server_caps();
+#if VALA_0_24
+      var caps = Notify.get_server_caps ();
+#else
+      var caps = notify_get_server_caps ();
+#endif
       bool persistence = false, actions = false;
-      foreach (string cap in caps) {
-        if (cap == "persistence")
-          persistence = true;
-        else if (cap == "actions")
-          actions = true;
-      }
+
+      persistence = (caps.find ("persistence") != null);
+      actions = (caps.find ("actions") != null);
+      
       if (persistence && actions) {
         // Ensure it's really Gnome-Shell, not a variation (like Cinnamon).
         string gsv = null;
