--- tools/assistant/compat/compat.pro	2010-10-08 16:54:09.804000000 +0300
+++ tools/assistant/compat/compat.pro	2010-10-08 16:54:37.884000000 +0300
@@ -14,6 +14,8 @@
 
 PROJECTNAME = Assistant
 
+DESTDIR = ../../../bin
+
 FORMS += helpdialog.ui \
         mainwindow.ui \
         tabbedbrowser.ui \
@@ -59,6 +61,9 @@
     TARGET = Assistant_adp
 }
 
+target.path=$$[QT_INSTALL_BINS]
+INSTALLS += target
+
 TRANSLATIONS = translations/assistant_adp_de.ts \
     translations/assistant_adp_ja.ts \
     translations/assistant_adp_pl.ts \
--- tools/assistant/assistant.pro	2010-09-10 12:04:47.000000000 +0300
+++ tools/assistant/assistant.pro	2010-06-02 05:03:09.000000000 +0300
@@ -3,4 +3,6 @@
 
 SUBDIRS += lib/fulltextsearch \
            lib \
-           tools
+           tools \
+           compat \
+           compat/lib \
\ No newline at end of file
--- /dev/null
+++ tools/assistant/compat/lib/QtAssistant/qassistantclient_global.h
@@ -0,0 +1 @@
+#include "../qassistantclient_global.h"
--- /dev/null	2010-09-28 21:07:22.854999850 +0300
+++ tools/assistant/compat/assistant.pro	2010-06-02 05:03:09.000000000 +0300
@@ -0,0 +1,84 @@
+include($$QT_SOURCE_TREE/tools/shared/fontpanel/fontpanel.pri)
+
+TEMPLATE = app
+LANGUAGE = C++
+TARGET = assistant_adp
+
+CONFIG += qt warn_on
+
+unix:contains(QT_CONFIG, dbus):QT += dbus
+
+build_all:!build_pass {
+    CONFIG -= build_all
+    CONFIG += release
+}
+QT += xml network
+
+PROJECTNAME = Assistant
+DESTDIR = ../../../bin
+
+FORMS += helpdialog.ui \
+        mainwindow.ui \
+        tabbedbrowser.ui \
+        topicchooser.ui
+
+SOURCES += main.cpp \
+        helpwindow.cpp \
+        topicchooser.cpp \
+        docuparser.cpp \
+        index.cpp \
+        profile.cpp \
+        config.cpp \
+        helpdialog.cpp \
+        mainwindow.cpp \
+        tabbedbrowser.cpp \
+        fontsettingsdialog.cpp
+
+HEADERS += helpwindow.h \
+        topicchooser.h \
+        docuparser.h \
+        index.h \
+        profile.h \
+        helpdialog.h \
+        mainwindow.h \
+        tabbedbrowser.h \
+        config.h \
+        fontsettingsdialog.h
+
+RESOURCES += assistant.qrc
+
+contains(QT_PRODUCT, OpenSource.*):DEFINES *= QT_OPENSOURCE
+DEFINES += QT_KEYWORDS
+#DEFINES +=  QT_PALMTOPCENTER_DOCS
+
+win32 {
+    !wince*:LIBS += -lshell32
+    RC_FILE = assistant.rc
+}
+
+mac {
+    ICON = assistant.icns
+    TARGET = Assistant_adp
+    QMAKE_INFO_PLIST = Info_mac.plist
+}
+
+target.path=$$[QT_INSTALL_BINS]
+INSTALLS += target
+
+TRANSLATIONS = assistant_de.ts
+
+unix:!contains(QT_CONFIG, zlib):LIBS += -lz
+
+contains(CONFIG, static): {
+    win32 {
+        exists($$[QT_INSTALL_PLUGINS]/imageformats/qjpeg.lib) {
+            QTPLUGIN += qjpeg
+            DEFINES += USE_STATIC_JPEG_PLUGIN            
+        }
+    } else {
+        exists($$[QT_INSTALL_PLUGINS]/imageformats/qjpeg.a) {
+            QTPLUGIN += qjpeg
+            DEFINES += USE_STATIC_JPEG_PLUGIN            
+        }        
+    }
+}
