diff -urN oxide-qt-1.16.5/build/config/compiler/BUILD.gn oxide-qt-1.16.5-patched/build/config/compiler/BUILD.gn
--- oxide-qt-1.16.5/build/config/compiler/BUILD.gn	2016-06-03 07:47:12.000000000 +0930
+++ oxide-qt-1.16.5-patched/build/config/compiler/BUILD.gn	2016-09-09 20:22:32.452104512 +0930
@@ -203,15 +203,6 @@
         cflags += [ "-fstack-protector" ]
       }
     }
-
-    # Linker warnings.
-    if (!(is_chromeos && current_cpu == "arm") &&
-        !(is_android && use_order_profiling) && !is_mac && !is_ios) {
-      # TODO(jochen): Enable this on chromeos on arm. http://crbug.com/356580
-      # TODO(lizeb,pasko): Fix link errors when linking with order_profiling=1
-      # crbug.com/485542
-      ldflags += [ "-Wl,--fatal-warnings" ]
-    }
   }
 
   if (is_clang && is_debug) {
