diff -Nur oxide-qt-1.2.5_orig/third_party/chromium/src/build/common.gypi oxide-qt-1.2.5/third_party/chromium/src/build/common.gypi
--- oxide-qt-1.2.5_orig/third_party/chromium/src/build/common.gypi	2014-09-03 14:43:14.000000000 +0200
+++ oxide-qt-1.2.5/third_party/chromium/src/build/common.gypi	2014-12-06 12:10:14.766461410 +0100
@@ -3315,23 +3315,6 @@
           ['win_use_allocator_shim==0', {
             'defines': ['NO_TCMALLOC'],
           }],
-          # _FORTIFY_SOURCE isn't really supported by Clang now, see
-          # http://llvm.org/bugs/show_bug.cgi?id=16821.
-          # It seems to work fine with Ubuntu 12 headers though, so use it
-          # in official builds.
-          ['os_posix==1 and (asan!=1 and msan!=1 and tsan!=1 and lsan!=1 and ubsan!=1) and (OS!="linux" or clang!=1 or buildtype=="Official")', {
-            'target_conditions': [
-              ['chromium_code==1', {
-                # Non-chromium code is not guaranteed to compile cleanly
-                # with _FORTIFY_SOURCE. Also, fortified build may fail
-                # when optimizations are disabled, so only do that for Release
-                # build.
-                'defines': [
-                  '_FORTIFY_SOURCE=2',
-                ],
-              }],
-            ],
-          }],
           ['OS=="linux" or OS=="android"', {
             'target_conditions': [
               ['_toolset=="target"', {
