From 80bc8ce643979fec201c4ebd5cd6405b6310357f Mon Sep 17 00:00:00 2001
From: Arx Cruz <arxcruz@gnome.org>
Date: Thu, 10 Oct 2013 19:52:38 +0000
Subject: But #702535 - List box doesn't expand to fill window

---
diff --git a/src/zenity.ui b/src/zenity.ui
index ee89093..7c0ab1f 100644
--- a/src/zenity.ui
+++ b/src/zenity.ui
@@ -1021,6 +1021,7 @@
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="shadow_type">in</property>
+                <property name="expand">True</property>
                 <child>
                   <object class="GtkTreeView" id="zenity_tree_view">
                     <property name="visible">True</property>
--
cgit v0.9.2
