diff -Naur indicator-datetime-13.10.0+14.04.20131217_orig/cmake/UseGSettings.cmake indicator-datetime-13.10.0+14.04.20131217/cmake/UseGSettings.cmake
--- indicator-datetime-13.10.0+14.04.20131217_orig/cmake/UseGSettings.cmake	2013-12-17 10:49:41.000000000 +0100
+++ indicator-datetime-13.10.0+14.04.20131217/cmake/UseGSettings.cmake	2013-12-31 09:06:17.424358665 +0100
@@ -10,14 +10,14 @@
     execute_process (COMMAND ${_glib_compile_schemas} --dry-run --schema-file=${SCHEMA_NAME} ERROR_VARIABLE _schemas_invalid OUTPUT_STRIP_TRAILING_WHITESPACE)
 
     if (_schemas_invalid)
-      message (SEND_ERROR "Schema validation error: ${_schemas_invalid}")
+      message (WARNING "Schema validation error: ${_schemas_invalid}")
     endif (_schemas_invalid)
 
     # Actually install and recomple schemas
     message (STATUS "${GSETTINGS_DIR} is the GSettings install dir")
     install (FILES ${SCHEMA_NAME} DESTINATION ${GSETTINGS_DIR} OPTIONAL)
 
-    install (CODE "message (STATUS \"Compiling GSettings schemas\")")
-    install (CODE "execute_process (COMMAND ${_glib_compile_schemas} ${GSETTINGS_DIR})")
+#    install (CODE "message (STATUS \"Compiling GSettings schemas\")")
+#    install (CODE "execute_process (COMMAND ${_glib_compile_schemas} ${GSETTINGS_DIR})")
 endmacro()
 
