diff --git a/gio/gdbus-2.0/codegen/gdbus-codegen.in b/gio/gdbus-2.0/codegen/gdbus-codegen.in
index 217c755..8520bb8 100644
--- a/gio/gdbus-2.0/codegen/gdbus-codegen.in
+++ b/gio/gdbus-2.0/codegen/gdbus-codegen.in
@@ -29,13 +29,9 @@ srcdir = os.getenv('UNINSTALLED_GLIB_SRCDIR', None)
 
 if srcdir is not None:
     path = os.path.join(srcdir, 'gio', 'gdbus-2.0')
-elif os.name == 'nt':
-    # Makes gdbus-codegen 'relocatable' at runtime on Windows.
-    path = os.path.join(os.path.dirname(__file__), '..', 'lib', 'gdbus-2.0')
+    sys.path.insert(0, os.path.abspath(path))
+    from codegen import codegen_main
 else:
-    path = os.path.join('@libdir@', 'gdbus-2.0')
-
-sys.path.insert(0, os.path.abspath(path))
-from codegen import codegen_main
+    from gdbus_codegen import codegen_main
 
 sys.exit(codegen_main.codegen_main())
