Index: b/src/main.c
===================================================================
--- a/src/main.c
+++ b/src/main.c
@@ -387,8 +387,10 @@
 {
 	GError * error = NULL;
 
+#if !GLIB_CHECK_VERSION (2, 35, 1)
 	/* Init the GTypes */
 	g_type_init();
+#endif
 
 	/* Setup i18n */
 	setlocale (LC_ALL, ""); 
Index: b/tests/dbus-interface.c
===================================================================
--- a/tests/dbus-interface.c
+++ b/tests/dbus-interface.c
@@ -571,7 +571,9 @@
 gint
 main (gint argc, gchar * argv[])
 {
+#if !GLIB_CHECK_VERSION (2, 35, 1)
 	g_type_init();
+#endif
 	g_test_init(&argc, &argv, NULL);
 
 	/* Test suites */
Index: b/tests/server-test.c
===================================================================
--- a/tests/server-test.c
+++ b/tests/server-test.c
@@ -270,7 +270,9 @@
 gint
 main (gint argc, gchar * argv[])
 {
+#if !GLIB_CHECK_VERSION (2, 35, 1)
 	g_type_init();
+#endif
 	g_test_init(&argc, &argv, NULL);
 
 	/* Test suites */
