diff -urN gnome-settings-daemon-3.32.0/meson.build gnome-settings-daemon-3.32.0-patched/meson.build
--- gnome-settings-daemon-3.32.0/meson.build	2019-03-12 09:13:44.000000000 +1030
+++ gnome-settings-daemon-3.32.0-patched/meson.build	2019-05-24 10:13:14.957354399 +0930
@@ -83,7 +83,6 @@
 add_project_arguments(common_flags + compiler_flags, language: 'c')
 
 glib_dep = dependency('glib-2.0', version: '>= 2.56')
-colord_dep = dependency('colord', version: '>= 1.0.2')
 geocode_glib_dep = dependency('geocode-glib-1.0', version: '>= 3.10.0')
 gio_dep = dependency('gio-2.0', version: '>= 2.53.0')
 gio_unix_dep = dependency('gio-unix-2.0')
@@ -127,6 +126,12 @@
   assert(enable_gudev, 'GUdev is not optional on Linux platforms')
 endif
 
+# Check for colord
+enable_colord = get_option('colord')
+if enable_colord
+  colord_dep = dependency('colord', version: '>= 1.0.2')
+endif
+
 has_timerfd_create = cc.has_function('timerfd_create')
 config_h.set10('HAVE_TIMERFD', has_timerfd_create)
 
@@ -139,8 +144,8 @@
 endif
 config_h.set10('HAVE_WAYLAND', enable_wayland)
 
-# wacom (disabled for s390/s390x and non Linux platforms)
-enable_wacom = host_is_linux_not_s390
+# wacom
+enable_wacom = get_option('wacom')
 if enable_wacom
   assert(enable_gudev, 'GUDev support is required for wacom support.')
   libwacom_dep = dependency('libwacom', version: '>= 0.7')
@@ -193,7 +198,6 @@
 
 # Sharing plugin
 enable_network_manager = get_option('network_manager')
-assert(enable_network_manager or not host_is_linux, 'NetworkManager support is not optional on Linux platforms')
 if enable_network_manager
   # network manager
   libnm_dep = dependency('libnm', version: '>= 1.0')
diff -urN gnome-settings-daemon-3.32.0/meson_options.txt gnome-settings-daemon-3.32.0-patched/meson_options.txt
--- gnome-settings-daemon-3.32.0/meson_options.txt	2019-03-12 09:13:44.000000000 +1030
+++ gnome-settings-daemon-3.32.0-patched/meson_options.txt	2019-05-24 10:00:13.836725988 +0930
@@ -3,8 +3,10 @@
 
 option('alsa', type: 'boolean', value: true, description: 'build with ALSA support (not optional on Linux platforms)')
 option('gudev', type: 'boolean', value: true, description: 'build with gudev device support (not optional on Linux platforms)')
+option('colord', type: 'boolean', value: true, description: 'build with colord support')
 option('cups', type: 'boolean', value: true, description: 'build with CUPS support')
 option('network_manager', type: 'boolean', value: true, description: 'build with NetworkManager support (not optional on Linux platforms)')
 option('rfkill', type: 'boolean', value: true, description: 'build with rfkill support (not optional on Linux platforms)')
 option('smartcard', type: 'boolean', value: true, description: 'build with smartcard support')
+option('wacom', type: 'boolean', value: true, description: 'build with Wacom devices support')
 option('wayland', type: 'boolean', value: true, description: 'build with Wayland support')
diff -urN gnome-settings-daemon-3.32.0/plugins/meson.build gnome-settings-daemon-3.32.0-patched/plugins/meson.build
--- gnome-settings-daemon-3.32.0/plugins/meson.build	2019-03-12 09:13:44.000000000 +1030
+++ gnome-settings-daemon-3.32.0-patched/plugins/meson.build	2019-05-24 10:00:45.123226380 +0930
@@ -1,7 +1,6 @@
 enabled_plugins = [
   ['a11y-settings', 'A11ySettings'],
   ['clipboard', 'Clipboard'],
-  ['color', 'Color'],
   ['datetime', 'Datetime'],
   ['dummy', ''],
   ['power', 'Power'],
@@ -15,6 +14,10 @@
   ['xsettings', 'XSettings']
 ]
 
+if enable_colord
+  enabled_plugins += [['color', 'Color']]
+endif
+
 if enable_smartcard
   enabled_plugins += [['smartcard', 'Smartcard']]
 endif
