diff -Nur a/meson.build b/meson.build
--- a/meson.build	2018-04-12 19:08:48.000000000 +0200
+++ b/meson.build	2018-05-29 23:11:29.665663309 +0200
@@ -134,7 +134,7 @@
 config_h.set10('HAVE_WAYLAND', enable_wayland)
 
 # wacom (disabled for s390/s390x and non Linux platforms)
-enable_wacom = host_is_linux_not_s390
+enable_wacom = host_is_linux_not_s390 and get_option('wacom')
 if enable_wacom
   assert(enable_gudev, 'GUDev support is required for wacom support.')
   libwacom_dep = dependency('libwacom', version: '>= 0.7')
diff -Nur a/meson_options.txt b/meson_options.txt
--- a/meson_options.txt	2018-04-12 19:08:48.000000000 +0200
+++ b/meson_options.txt	2018-05-29 23:14:10.566310613 +0200
@@ -7,4 +7,5 @@
 option('network_manager', type: 'boolean', value: true, description: 'build with NetworkManager support (not optional on Linux platforms)')
 option('rfkill', type: 'boolean', value: true, description: 'build with rfkill support (not optional on Linux platforms)')
 option('smartcard', type: 'boolean', value: true, description: 'build with smartcard support')
+option('wacom', type: 'boolean', value: true, description: 'build with Wacom support')
 option('wayland', type: 'boolean', value: true, description: 'build with Wayland support')
diff -Nur a/plugins/common/gsd-device-manager-udev.c b/plugins/common/gsd-device-manager-udev.c
--- a/plugins/common/gsd-device-manager-udev.c	2018-04-12 19:08:48.000000000 +0200
+++ b/plugins/common/gsd-device-manager-udev.c	2018-05-29 21:58:44.670661727 +0200
@@ -24,7 +24,9 @@
 #include <string.h>
 #include <gudev/gudev.h>
 
+#if HAVE_WAYLAND
 #include <gdk/gdkwayland.h>
+#endif
 #include "gsd-device-manager-udev.h"
 
 struct _GsdUdevDeviceManager
@@ -248,8 +250,10 @@
 	GHashTableIter iter;
 	GsdDevice *device;
 
+#if HAVE_WAYLAND
 	node_path = gdk_wayland_device_get_node_path (gdk_device);
 	if (!node_path)
+#endif
 		return NULL;
 
 	g_hash_table_iter_init (&iter, GSD_UDEV_DEVICE_MANAGER (manager)->devices);
