diff -urN compiz-0.9.8+bzr3249/plugins/decor/decor.xml.in compiz-0.9.8+bzr3249-patched/plugins/decor/decor.xml.in
--- compiz-0.9.8+bzr3249/plugins/decor/decor.xml.in	2012-08-16 01:04:17.000000000 +0930
+++ compiz-0.9.8+bzr3249-patched/plugins/decor/decor.xml.in	2012-08-22 14:37:57.018312648 +0930
@@ -109,7 +109,7 @@
 	    <option name="decoration_match" type="match">
 		<_short>Decoration windows</_short>
 		<_long>Windows that should be decorated</_long>
-		<default>any</default>
+		<default>!state=maxvert</default>
 	    </option>
 	    <option name="shadow_match" type="match">
 		<_short>Shadow windows</_short>
