diff -urN compiz-0.9.8+bzr3249/cmake/base.cmake compiz-0.9.8+bzr3249-patched//cmake/base.cmake
--- compiz-0.9.8+bzr3249/cmake/base.cmake	2012-06-30 01:36:55.000000000 +0930
+++ compiz-0.9.8+bzr3249-patched//cmake/base.cmake	2012-08-07 08:23:57.966910134 +0930
@@ -48,13 +48,9 @@
         message ("Call \"sudo make findcompiz_install\" to install it.\n")
         compiz_print_configure_footer ()
     endif ()
-    install (FILES
-	     ${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindCompiz.cmake
-	     DESTINATION
-	     ${CMAKE_INSTALL_PREFIX}/share/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/)
     add_custom_target (findcompiz_install
-	${CMAKE_COMMAND} -E make_directory ${CMAKE_ROOT}/Modules &&
-	${CMAKE_COMMAND} -E copy ${CMAKE_SOURCE_DIR}/cmake/FindCompiz.cmake ${CMAKE_ROOT}/Modules
+	${CMAKE_COMMAND} -E make_directory ${COMPIZ_DESTDIR}${CMAKE_ROOT}/Modules &&
+	${CMAKE_COMMAND} -E copy ${CMAKE_SOURCE_DIR}/cmake/FindCompiz.cmake ${COMPIZ_DESTDIR}${CMAKE_ROOT}/Modules
     )
     find_file (_find_compizconfig FindCompizConfig.cmake PATHS ${CMAKE_ROOT}/Modules ${ARGN})
     if (NOT _find_compizconfig)
@@ -64,14 +60,10 @@
 	message ("Call \"sudo make findcompiz_install\" to install it.\n")
 	compiz_print_configure_footer ()
     endif (NOT _find_compizconfig)
-    install (FILES
-	     ${CMAKE_CURRENT_SOURCE_DIR}/compizconfig/libcompizconfig/cmake/FindCompizConfig.cmake
-	     DESTINATION
-	     ${CMAKE_INSTALL_PREFIX}/share/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/)
     add_custom_target (
 	findcompizconfig_install
-	${CMAKE_COMMAND} -E make_directory ${CMAKE_ROOT}/Modules &&
-	${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/compizconfig/libcompizconfig/cmake/FindCompizConfig.cmake ${CMAKE_ROOT}/Modules
+	${CMAKE_COMMAND} -E make_directory ${COMPIZ_DESTDIR}${CMAKE_ROOT}/Modules &&
+	${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/compizconfig/libcompizconfig/cmake/FindCompizConfig.cmake ${COMPIZ_DESTDIR}${CMAKE_ROOT}/Modules
     )
 endfunction ()
 
