Same as the 2.8 version, except removed gdk-pixbuf parts

--- gtk+-2.21.3.orig/gtk/gtkrc.c
+++ gtk+-2.21.3/gtk/gtkrc.c
@@ -364,7 +364,15 @@
       if (im_module_file)
 	result = g_strdup (im_module_file);
       else
-	result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules", NULL);
+	{
+	  result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", GTK_HOST, "gtk.immodules", NULL);
+	  if (!g_file_test (result, G_FILE_TEST_EXISTS))
+	    {
+	      g_free (result);
+	      
+	      result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules", NULL);
+	    }
+	}
     }
 
   return result;
