diff -Naur indicator-sound-12.10.2+14.04.20131125_orig/src/service.vala indicator-sound-12.10.2+14.04.20131125/src/service.vala
--- indicator-sound-12.10.2+14.04.20131125_orig/src/service.vala	2013-11-25 04:55:59.000000000 +0100
+++ indicator-sound-12.10.2+14.04.20131125/src/service.vala	2013-12-30 10:13:00.948291102 +0100
@@ -17,6 +17,11 @@
  *      Lars Uebernickel <lars.uebernickel@canonical.com>
  */
 
+#if !VALA_0_24
+	[CCode (cheader_filename = "libnotify/notify.h", cname = "notify_get_server_caps")]
+	extern static GLib.List<string> notify_get_server_caps ();
+#endif
+
 public class IndicatorSound.Service {
 	public Service () {
 		this.settings = new Settings ("com.canonical.indicator.sound");
@@ -48,7 +53,11 @@
 		});
 
 		if (settings.get_boolean ("show-notify-osd-on-scroll")) {
-			unowned List<string> caps = Notify.get_server_caps ();
+#if VALA_0_24
+			var caps = Notify.get_server_caps ();
+#else
+			var caps = notify_get_server_caps ();
+#endif
 			if (caps.find_custom ("x-canonical-private-synchronous", strcmp) != null) {
 				this.notification = new Notify.Notification ("indicator-sound", "", "");
 				this.notification.set_hint_string ("x-canonical-private-synchronous", "indicator-sound");
