From 3fe83b83f422d08f0030195f4951b332706ad7f8 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Mon, 25 Jun 2012 01:48:39 -0400
Subject: [PATCH] Additional Gentoo system user accounts with non-trivial
 login shell

---
 src/daemon.c |   34 ++++++++++++++++++++++++++++++++++
 1 files changed, 34 insertions(+), 0 deletions(-)

diff --git a/src/daemon.c b/src/daemon.c
index d1708ca..2ad2aff 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -77,6 +77,40 @@ static const char *default_excludes[] = {
         "games",
         "man",
         "at",
+        /* Additional Gentoo system users with non-trivial login shell */
+        "amanda",
+        "backuppc",
+        "drqueue",
+        "firebird",
+        "flexlm",
+        "foldingathome",
+        "geneweb",
+        "git",
+        "gnump3d",
+        "hacluster",
+        "hg",
+        "hsqldb",
+        "infinote",
+        "jffnms",
+        "klive",
+        "mailman",
+        "mpd",
+        "mythtv",
+        "nagios",
+        "nx",
+        "oneadmin",
+        "openvpn",
+        "p2p",
+        "phxd",
+        "resin",
+        "rplayd",
+        "scponly",
+        "secoff",
+        "tinyproxy",
+        "ttrssd",
+        "vboxguest",
+        "vdr",
+        "vdradmin",
         NULL
 };
 
-- 
1.7.8.6

