diff -urN nux-4.0.2+13.10.20130829.1/NuxGraphics/GraphicsDisplayX11.cpp nux-4.0.2+13.10.20130829.1-patched/NuxGraphics/GraphicsDisplayX11.cpp
--- nux-4.0.2+13.10.20130829.1/NuxGraphics/GraphicsDisplayX11.cpp	2013-09-03 08:36:11.887170623 +0930
+++ nux-4.0.2+13.10.20130829.1-patched/NuxGraphics/GraphicsDisplayX11.cpp	2013-09-03 08:36:28.679397097 +0930
@@ -1724,14 +1724,13 @@
         if (!skip)
         {
           int num_char_stored = 0;
-
           if (m_xim_controller->IsXICValid())
           {
             delete[] m_pEvent->dtext;
             m_pEvent->dtext = nullptr;
 
-            num_char_stored = XmbLookupString(m_xim_controller->GetXIC(), &xevent.xkey, buffer,
-                                              NUX_EVENT_TEXT_BUFFER_SIZE, (KeySym*) &m_pEvent->x11_keysym, nullptr);
+            num_char_stored = XmbLookupString(m_xim_controller->GetXIC(), &xevent.xkey, nullptr,
+                                              0, (KeySym*) &m_pEvent->x11_keysym, nullptr);
 
             if (num_char_stored > 0)
             {
@@ -1755,12 +1754,6 @@
           }
         }
 
-        if (m_pEvent->dtext == nullptr)
-        {
-          m_pEvent->dtext = new char[NUX_EVENT_TEXT_BUFFER_SIZE];
-          m_pEvent->dtext[0] = 0;
-        }
-
         break;
       }
 
