Index: lightdm-1.0.6/data/lightdm.conf
===================================================================
--- lightdm-1.0.6.orig/data/lightdm.conf
+++ lightdm-1.0.6/data/lightdm.conf
@@ -65,7 +65,7 @@
 #user-session=default
 #allow-guest=true
 #guest-session=UNIMPLEMENTED
-#session-wrapper=lightdm-session
-#display-setup-script=
+session-wrapper=/etc/lightdm/Xsession
+display-setup-script=/etc/lightdm/lightdm-greeter-display-setup
 #greeter-setup-script=
 #session-setup-script=
