diff -urN libcolumbus-1.1.0+14.04.20140325.3/cmake/python.cmake libcolumbus-1.1.0+14.04.20140325.3-patched/cmake/python.cmake
--- libcolumbus-1.1.0+14.04.20140325.3/cmake/python.cmake	2014-03-26 06:06:00.000000000 +1030
+++ libcolumbus-1.1.0+14.04.20140325.3-patched/cmake/python.cmake	2014-12-08 04:13:11.186915742 +1030
@@ -10,7 +10,7 @@
 if(use_python2)
   pkg_search_module(PYTHONLIBS python)
 else()
-  pkg_search_module(PYTHONLIBS python3)
+  find_package(PythonLibs)
 endif()
 
 if(NOT BP_HEADER)
@@ -19,13 +19,13 @@
   if(NOT PYTHONLIBS_FOUND)
     message(STATUS "Python dev libraries not found, not building Python bindings.")
   else()
-    string(SUBSTRING ${PYTHONLIBS_VERSION} 0 1 PYTHON_MAJOR)
-    string(SUBSTRING ${PYTHONLIBS_VERSION} 2 1 PYTHON_MINOR)
-    message(STATUS "Found Python version ${PYTHON_MAJOR}.${PYTHON_MINOR}.")
+    string(SUBSTRING ${PYTHONLIBS_VERSION_STRING} 0 1 PYTHON_MAJOR)
+    string(SUBSTRING ${PYTHONLIBS_VERSION_STRING} 2 1 PYTHON_MINOR)
+    message(STATUS "Found Python version ${PYTHON_MAJOR}.${PYTHON_MINOR}")
     if(NOT use_python2)
       execute_process(COMMAND ${CMAKE_SOURCE_DIR}/cmake/pysoabi.py OUTPUT_VARIABLE pysoabi OUTPUT_STRIP_TRAILING_WHITESPACE)
     endif()
-    find_library(BOOST_PYTHON_HACK boost_python-py${PYTHON_MAJOR}${PYTHON_MINOR})
+    find_library(BOOST_PYTHON_HACK boost_python-${PYTHON_MAJOR}.${PYTHON_MINOR})
 
     if(NOT BOOST_PYTHON_HACK)
       message(STATUS "Boost.Python hack library not found, not building Python bindings")
