From 7e8f45e66233e502aa9ae5a3f5200884b1b9b235 Mon Sep 17 00:00:00 2001
From: Alessio Treglia <alessio@debian.org>
Date: Thu, 21 Dec 2017 15:05:01 +1300
Subject: [PATCH 03/12] Replace useradd with adduser to comply with Debian
 Policy.

---
 src/daemon.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/daemon.c b/src/daemon.c
index 5f2931e..140210c 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -1106,12 +1106,13 @@ daemon_create_user_authorized_cb (Daemon                *daemon,
 
         sys_log (context, "create user '%s'", cd->user_name);
 
-        argv[0] = "/usr/sbin/useradd";
-        argv[1] = "-m";
-        argv[2] = "-c";
-        argv[3] = cd->real_name;
-        argv[4] = cd->user_name;
-        argv[5] = NULL;
+        argv[0] = "/usr/sbin/adduser";
+        argv[1] = "--quiet";
+        argv[2] = "--disabled-login";
+        argv[3] = "--gecos";
+        argv[4] = cd->real_name;
+        argv[5] = cd->user_name;
+        argv[6] = NULL;
 
         if (!spawn_with_login_uid (context, argv, &error)) {
                 throw_error (context, ERROR_FAILED, "running '%s' failed: %s", argv[0], error->message);
-- 
2.20.1

